From 6e90e92480b78bea92102398d9eda4089d3d5da9 Mon Sep 17 00:00:00 2001
From: linmin <linmin@eswincomputing.com>
Date: Mon, 5 Aug 2024 19:34:14 +0800
Subject: [PATCH 244/442] WIN2030-14758:feat:Add eic7702-evb-a1.dts

Changelogs:
1.Add eic7702-evb-a1.dts
2.Modified eswin-win2030-arch-d2d.dtsi:remove the nodes of die0, include the
  eswin-win2030-arch.dtsi instead.
3.Add configs/eic7702_defconfig for dual die configuration

Signed-off-by: linmin <linmin@eswincomputing.com>
Change-Id: If476a6b011b54c3be01acfe65cdce873135e241e
---
 arch/riscv/boot/dts/eswin/Makefile            |    3 +-
 arch/riscv/boot/dts/eswin/eic7702-evb-a1.dts  | 1242 +++++++++++++++++
 .../dts/eswin/eswin-win2030-arch-d2d.dtsi     | 1171 +++++-----------
 .../boot/dts/eswin/eswin-win2030-arch.dtsi    |   71 +-
 .../dts/eswin/eswin-win2030-die0-soc.dtsi     |    5 +-
 .../dts/eswin/eswin-win2030-die1-soc.dtsi     |   16 +-
 .../dts/eswin/eswin-win2030-platform.dtsi     |    2 +-
 arch/riscv/boot/dts/eswin/eswin-win2030.dts   |    4 +-
 arch/riscv/configs/eic7702_defconfig          |  834 +++++++++++
 9 files changed, 2469 insertions(+), 879 deletions(-)
 create mode 100644 arch/riscv/boot/dts/eswin/eic7702-evb-a1.dts
 create mode 100644 arch/riscv/configs/eic7702_defconfig

diff --git a/arch/riscv/boot/dts/eswin/Makefile b/arch/riscv/boot/dts/eswin/Makefile
index 8c431bb594ea..83ba4023ed5e 100644
--- a/arch/riscv/boot/dts/eswin/Makefile
+++ b/arch/riscv/boot/dts/eswin/Makefile
@@ -6,6 +6,7 @@ dtb-$(CONFIG_SOC_SIFIVE) += eswin-win2030.dtb \
                                                         eic7700-milkv-megrez.dtb \
                                                         eic7700-pine64-starpro64.dtb \
 							eic7702-evb-a1-d0.dtb \
-							eic7702-evb-a1-d1.dtb
+							eic7702-evb-a1-d1.dtb \
+							eic7702-evb-a1.dtb
 
 obj-$(CONFIG_BUILTIN_DTB) += $(addsuffix .o, $(dtb-y))
diff --git a/arch/riscv/boot/dts/eswin/eic7702-evb-a1.dts b/arch/riscv/boot/dts/eswin/eic7702-evb-a1.dts
new file mode 100644
index 000000000000..a388e009f877
--- /dev/null
+++ b/arch/riscv/boot/dts/eswin/eic7702-evb-a1.dts
@@ -0,0 +1,1242 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Device Tree file for Eswin EIC7700 SoC.
+ *
+ * Copyright 2024, Beijing ESWIN Computing Technology Co., Ltd.. All rights reserved.
+ * SPDX-License-Identifier: GPL-2.0
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/dts-v1/;
+
+#define RTCCLK_FREQ		1000000
+#define LSPCLK_FREQ		200000000
+
+/* If wanna enable ECC capability of DDR, should reserve highest zone of 1/8 all space for it */
+#define MEMORY_SIZE_H		0x4
+#define MEMORY_SIZE_L		0x0
+#define CMA_SIZE		0x20000000
+
+#include "eswin-win2030-die0-soc.dtsi"
+#include "eswin-win2030-die1-soc.dtsi"
+
+#include "eic7700-pinctrl.dtsi"
+#include <dt-bindings/interrupt-controller/irq.h>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/* Clock frequency (in Hz) of the PCB crystal for rtcclk */
+
+/ {
+	#address-cells = <2>;
+	#size-cells = <2>;
+	model = "ESWIN EIC7700";
+	compatible = "sifive,hifive-unmatched-a00", "sifive,fu740-c000",
+		     "sifive,fu740", "eswin,eic7700";
+
+	aliases {
+		serial0 = &d0_uart0;
+		ethernet0 = &d0_gmac0;
+		ethernet1 = &d0_gmac1;
+		rtc0 = &die0_rtc;
+		rtc1 = &pcf_rtc;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	cpus {
+		timebase-frequency = <RTCCLK_FREQ>;
+	};
+
+	memory@59000000 {
+		device_type = "memory";
+		reg = <0x0 0x59000000 0x0 0x400000>;
+		numa-node-id = <0>;
+	};
+
+	memory@79000000 {
+		device_type = "memory";
+		reg = <0x0 0x79000000 0x0 0x400000>;
+		numa-node-id = <1>;
+	};
+
+	memory@80000000 {
+		device_type = "memory";
+		reg = <0x0 0x80000000 MEMORY_SIZE_H MEMORY_SIZE_L>;
+		numa-node-id = <0>;
+	};
+
+	memory@2000000000 {
+		device_type = "memory";
+		reg = <0x20 0x00000000 MEMORY_SIZE_H MEMORY_SIZE_L>;
+		numa-node-id = <1>;
+	};
+
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		linux,cma {
+			compatible = "shared-dma-pool";
+			reusable;
+			size = <0x0 CMA_SIZE>;
+			alignment = <0x0 0x1000>;
+			alloc-ranges = <0x20 0x00000000 MEMORY_SIZE_H MEMORY_SIZE_L>;
+			linux,cma-default;
+		};
+
+		npu0_reserved: sprammemory@59000000 {
+			no-map;
+			reg = <0x0 0x59000000 0x0 0x400000>;
+		};
+
+		npu1_reserved: sprammemory@79000000 {
+			no-map;
+			reg = <0x0 0x79000000 0x0 0x400000>;
+		};
+
+		g2d_4GB_boundary_reserved_4k {
+			no-map;
+			reg = <0x0 0xfffff000 0x0 0x1000>;
+		};
+
+		g2d_8GB_boundary_reserved_4k {
+			no-map;
+			reg = <0x1 0xfffff000 0x0 0x1000>;
+		};
+
+		g2d_12GB_boundary_reserved_4k {
+			no-map;
+			reg = <0x2 0xfffff000 0x0 0x1000>;
+		};
+
+		mmz_nid_0_part_0 {
+			compatible = "eswin-reserve-memory";
+			reg = <0x3 0x0 0x1 0x80000000>;
+			no-map;
+		};
+
+		d1_g2d_4GB_boundary_reserved_4k {
+			no-map;
+			reg = <0x20 0xfffff000 0x0 0x1000>;
+		};
+
+		d1_g2d_8GB_boundary_reserved_4k {
+			no-map;
+			reg = <0x21 0xfffff000 0x0 0x1000>;
+		};
+
+		d1_g2d_12GB_boundary_reserved_4k {
+			no-map;
+			reg = <0x22 0xfffff000 0x0 0x1000>;
+		};
+
+		mmz_nid_1_part_0 {
+			compatible = "eswin-reserve-memory";
+			reg = <0x22 0x80000000 0x1 0x80000000>;
+			no-map;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_gpio107_default &pinctrl_gpio108_default &pinctrl_gpio109_default>;
+
+		gpio-107 {
+			gpios = <&portd 11 GPIO_ACTIVE_HIGH>;
+			label = "power";
+			linux,default-trigger = "default-on";
+		};
+
+		gpio-108 {
+			gpios = <&portd 12 GPIO_ACTIVE_LOW>;
+			label = "heartbeat";
+			linux,default-trigger = "heartbeat";
+		};
+
+		gpio-109 {
+			gpios = <&portd 13 GPIO_ACTIVE_HIGH>;
+			label = "gpio-109";
+			linux,default-trigger = "default-off";
+		};
+
+		gpio-110 {
+			gpios = <&portd 14 GPIO_ACTIVE_HIGH>;
+			label = "gpio-109";
+			linux,default-trigger = "default-off";
+		};
+	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+
+		ok_key {
+			label = "OK";
+			linux,code = <KEY_OK>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&pinctrl_gpio6_default>;
+			gpios = <&porta 6 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&d0_clock {
+	status = "okay";
+};
+&d1_clock {
+	status = "okay";
+};
+
+&d0_reset {
+	status = "okay";
+};
+&d1_reset {
+	status = "okay";
+};
+
+&d0_pmu {
+	status = "okay";
+};
+&d1_pmu {
+	status = "disabled";
+};
+
+&ddr0 {
+	status = "okay";
+};
+
+&ddr1 {
+	status = "okay";
+};
+
+&smmu0 {
+	status = "okay";
+};
+&smmu1 {
+	status = "okay";
+};
+
+&smmu_pmu0 {
+	status = "disabled";
+};
+&smmu_pmu1 {
+	status = "disabled";
+};
+
+
+&dev_foo_a {
+	status = "okay";
+};
+
+&d0_cfg_noc {
+	status = "okay";
+};
+
+&d0_llc_noc {
+	status = "okay";
+	stat,0 = "TracePort:ddr0_p0_req";
+	stat,1 = "TracePort:ddr1_p0_req";
+	//latency,0 = "TracePort:llcnoc_trans_probe";
+	//pending,0 = "TracePort:llcnoc_trans_probe";
+};
+
+&d0_sys_noc {
+	status = "okay";
+
+	//eswin,DSPT-qos-owner;
+	//eswin,NPU-qos-owner;
+	//eswin,SPISLV_TBU3-qos-owner;
+
+	stat,0 = "TracePort:ddr0_p1_req",
+		 "InitFlow:mcput_snoc_mp/I/0";
+
+	stat,1 = "TracePort:ddr0_p2_req",
+		 "InitFlow:dspt_snoc/I/0",
+		 "AddrBase:0x81000000", "AddrSize:0x30",
+		 "Opcode:RdWrLockUrg", "Status:ReqRsp", "Length:0x8000", "Urgency:0x0";
+
+	stat,2 = "TracePort:ddr1_p1_req",
+		 "Status:Req", "AddrSize:0x28";
+
+	stat,3 = "TracePort:ddr1_p2_req";
+
+	latency,0 = "TracePort:sysnoc_trans_probe_0", "AddrSize:0x0";
+	latency,1 = "TracePort:sysnoc_trans_probe_1","Mode:latency","AddrBase:0x82000000","AddrSize:0x28","Opcode:RdWr";
+	//latency,2 = "TracePort:sysnoc_trans_probe_2";
+
+	//pending,0 = "TracePort:sysnoc_trans_probe_0";
+	//pending,1 = "TracePort:sysnoc_trans_probe_1","Mode:latency","AddrBase:0x82000000","AddrSize:0x0","Opcode:RdWr";
+	pending,0 = "TracePort:sysnoc_trans_probe_2", "AddrSize:0x3";
+};
+
+&d0_media_noc {
+	status = "okay";
+
+	//eswin,GPU-qos-owner;
+	//eswin,TBU2-qos-owner;
+	//eswin,VC-qos-owner;
+
+	stat,0 = "TracePort:ddr0_p3_req";
+	stat,1 = "TracePort:ddr1_p3_req";
+	//latency,0 = "TracePort:mnoc_trans_probe";
+	//pending,0 = "TracePort:mnoc_trans_probe";
+};
+
+&d0_realtime_noc {
+	status = "okay";
+
+	//eswin,TBU0-qos-owner;
+	//eswin,VO-qos-owner;
+
+	stat,0 = "TracePort:ddr0_p4_req";
+	stat,1 = "TracePort:ddr1_p4_req";
+	//latency,0 = "TracePort:rnoc_trans_probe";
+	//pending,0 = "TracePort:rnoc_trans_probe";
+};
+
+&d0_noc_wdt {
+	status = "okay";
+};
+
+&d0_ipc_scpu {
+	status = "okay";
+};
+&d1_ipc_scpu {
+	status = "disabled";
+};
+
+&d0_lpcpu {
+	status = "okay";
+};
+&d1_lpcpu {
+	status = "disabled";
+};
+
+&pcie {
+	status = "okay";
+};
+&d0_pcie {
+	status = "disabled";
+};
+
+&d0_npu{
+	status = "okay";
+};
+&d1_npu {
+	status = "disabled";
+};
+
+&d0_dsp_subsys {
+	status = "okay";
+};
+&d1_dsp_subsys {
+	status = "disabled";
+};
+
+&d0_dsp0 {
+	status = "okay";
+};
+&d1_dsp0 {
+	status = "disabled";
+};
+
+&d0_dsp1 {
+	status = "okay";
+};
+&d1_dsp1 {
+	status = "disabled";
+};
+
+&d0_dsp2 {
+	status = "okay";
+};
+&d1_dsp2 {
+	status = "disabled";
+};
+
+&d0_dsp3 {
+	status = "okay";
+};
+&d1_dsp2 {
+	status = "disabled";
+};
+
+&gpu0 {
+	status = "okay";
+};
+&d1_gpu {
+	status = "disabled";
+};
+
+&gc820 {
+	status = "okay";
+};
+&d1_gc820 {
+	status = "disabled";
+};
+
+&vdec0 {
+	status = "okay";
+};
+&vdec1 {
+	status = "disabled";
+};
+
+&venc0 {
+	status = "okay";
+};
+&venc1 {
+	status = "disabled";
+};
+
+&video_output {
+	status = "okay";
+};
+&d1_video_output {
+	status = "disabled";
+};
+
+&dc {
+	status = "okay";
+};
+&d1_dc {
+	status = "disabled";
+};
+
+&dc_test {
+	status = "disabled";
+};
+&d1_dc_test {
+	status = "disabled";
+};
+
+&virtual_display {
+	status = "okay";
+};
+&d1_virtual_display {
+	status = "disabled";
+};
+
+&dsi_output {
+	status = "okay";
+};
+&d1_dsi_output {
+	status = "disabled";
+};
+
+&dsi_controller {
+    status = "okay";
+};
+&d1_dsi_controller {
+	status = "disabled";
+};
+
+&dsi_panel {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio27_default &pinctrl_gpio106_default>;
+	backlight0-gpios = <&porta 27 GPIO_ACTIVE_HIGH>;
+	rst-gpios = <&portd 10 GPIO_ACTIVE_HIGH>;
+};
+
+&dw_hdmi {
+	status = "okay";
+	ports {
+		port@2 {
+			reg = <2>;
+			hdmi_in_i2s: endpoint@1 {
+				system-clock-frequency = <12288000>;
+				remote-endpoint = <&d0_i2s0_endpoint>;
+			};
+		};
+	};
+};
+&d1_dw_hdmi {
+	status = "disabled";
+};
+
+&dw_hdmi_hdcp2 {
+	status = "okay";
+};
+&d1_dw_hdmi_hdcp2 {
+	status = "disabled";
+};
+
+
+&d0_i2s0 {
+	status = "okay";
+	d0_i2s0_port: port {
+		d0_i2s0_endpoint: endpoint {
+			remote-endpoint = <&hdmi_in_i2s>;
+			dai-format = "i2s";
+		};
+	};
+};
+&d1_i2s0 {
+	status = "disabled";
+};
+
+&d0_i2s1 {
+	status = "okay";
+	d0_i2s1_port: port {
+		d0_i2s1_endpoint: endpoint {
+			remote-endpoint = <&d0_codec0_endpoint>;
+			dai-format = "i2s";
+		};
+	};
+};
+&d1_i2s1 {
+	status = "disabled";
+};
+
+&d0_i2s2 {
+	status = "okay";
+	d0_i2s2_port: port {
+		d0_i2s2_endpoint: endpoint {
+			remote-endpoint = <&d0_codec1_endpoint>;
+			dai-format = "i2s";
+		};
+	};
+};
+&d1_i2s2 {
+	status = "disabled";
+};
+
+&d0_graphcard0 {
+	status = "okay";
+	label = "Analog Audio-0";
+	dais = <&d0_i2s1_port>;
+};
+
+&d0_graphcard1 {
+	status = "okay";
+	label = "Analog Audio-1";
+	dais = <&d0_i2s2_port>;
+};
+
+&d0_graphcard2 {
+	status = "okay";
+	label = "HDMI Audio";
+	dais = <&d0_i2s0_port>;
+};
+
+&d1_graphcard {
+	status = "disabled";
+};
+
+&isp_0 {
+	status = "okay";
+};
+&d1_isp_0 {
+	status = "disabled";
+};
+
+&isp_1 {
+	status = "okay";
+};
+&d1_isp_1 {
+	status = "disabled";
+};
+
+&dewarp {
+	status = "okay";
+};
+&d1_dewarp {
+	status = "disabled";
+};
+
+&mipi_dphy_rx {
+	status = "okay";
+};
+&d1_mipi_dphy_rx {
+	status = "disabled";
+};
+
+&csi_dma0 {
+	status = "okay";
+};
+&d1_csi_dma0 {
+	status = "disabled";
+};
+
+&csi_dma1 {
+	status = "disabled";
+};
+&d1_csi_dma1 {
+	status = "disabled";
+};
+
+&csi2_0 {
+	status = "okay";
+};
+&d1_csi2_0 {
+	status = "disabled";
+};
+
+&csi2_1 {
+	status = "disabled";
+};
+&d1_csi2_1 {
+	status = "disabled";
+};
+
+&sdhci_emmc {
+	/* emmc */
+	status = "okay";
+	delay_code = <0x17>;
+	drive-impedance-ohm = <50>;
+	enable-cmd-pullup;
+	enable-data-pullup;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_emmc_led_control_default>;
+	no-sdio;
+	no-sd;
+};
+&d1_sdhci_emmc {
+	status = "disabled";
+};
+
+&sdio0 {
+	/* sd card */
+	status = "okay";
+	delay_code = <0x29>;
+	drive-impedance-ohm = <33>;
+	enable-cmd-pullup;
+	enable-data-pullup;
+	no-sdio;
+	no-mmc;
+};
+&d1_sdio0 {
+	status = "disabled";
+};
+
+&sdio1 {
+	/* wifi module */
+	status = "okay";
+	delay_code = <0x29>;
+	drive-impedance-ohm = <33>;
+	enable-cmd-pullup;
+	enable-data-pullup;
+	keep-power-in-suspend;
+	non-removable;
+	no-sd;
+	no-mmc;
+};
+&d1_sdio1 {
+	status = "disabled";
+};
+
+&d0_gmac0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio15_default>;
+	rst-gpios = <&porta 15 GPIO_ACTIVE_LOW>;
+	eswin,rgmiisel = <&pinctrl 0x290 0x3>;
+	eswin,led-cfgs = <0x6100 0xa40 0x420>;
+
+	status = "okay";
+};
+&d1_gmac0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio13_default>;
+	rst-gpios = <&porta 13 GPIO_ACTIVE_LOW>;
+	eswin,rgmiisel = <&pinctrl 0x290 0x3>;
+	eswin,led-cfgs = <0x6100 0xa40 0x420>;
+
+	status = "okay";
+};
+
+&d0_gmac1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio14_default>;
+	rst-gpios = <&porta 14 GPIO_ACTIVE_LOW>;
+	eswin,rgmiisel = <&pinctrl 0x294 0x3>;
+	eswin,led-cfgs = <0x6100 0xa40 0x420>;
+
+	status = "okay";
+};
+&d1_gmac1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio14_default>;
+	rst-gpios = <&porta 14 GPIO_ACTIVE_LOW>;
+	eswin,rgmiisel = <&pinctrl 0x294 0x3>;
+	eswin,led-cfgs = <0x6100 0xa40 0x420>;
+
+	status = "okay";
+};
+
+
+&d0_sata {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_sata_act_led_default>;
+};
+&d1_sata {
+	status = "disabled";
+};
+
+&d0_usbdrd3_0 {
+	status = "okay";
+};
+&d1_usbdrd3_0 {
+	status = "disabled";
+};
+
+&d0_usbdrd_dwc3_0 {
+	status = "okay";
+	dr_mode = "host";
+	maximum-speed = "super-speed";
+};
+
+&d0_usbdrd3_1 {
+	status = "okay";
+};
+&d1_usbdrd3_0 {
+	status = "disabled";
+};
+
+&d0_usbdrd_dwc3_1 {
+	status = "okay";
+	dr_mode = "host";
+	maximum-speed = "super-speed";
+};
+&d1_usbdrd_dwc3_1 {
+	status = "disabled";
+};
+
+&d0_dmac0 {
+	status = "okay";
+};
+&d1_dmac0 {
+	status = "disabled";
+};
+
+&d0_aon_dmac {
+	status = "okay";
+};
+&d1_aon_dmac {
+	status = "okay";
+};
+
+&d0_uart0 {
+	/* debug */
+	status = "okay";
+};
+&d1_uart0 {
+	status = "okay";
+};
+
+&d0_uart1 {
+	/* BT M.2 KEY-E */
+	status = "okay";
+};
+&d1_uart1 {
+	status = "okay";
+};
+
+&d0_uart2 {
+	/* RS232 DB9 */
+	status = "okay";
+};
+
+&d0_uart3 {
+	/* unused */
+	status = "disabled";
+};
+
+&d0_uart4 {
+	/* unused */
+	status = "disabled";
+};
+
+&ssi0 {
+	/* spi flash */
+	status = "okay";
+	num-cs = <2>;
+	spi-flash@0 {
+		compatible = "winbond,w25q128fw",
+					"jedec,spi-nor";
+		reg = <0>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		spi-max-frequency = <4800000>;
+		rx-sample-delay-ns = <10>;
+	};
+	spi-flash@1 {
+		compatible = "winbond,w25q128fw",
+					"jedec,spi-nor";
+		reg = <1>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		spi-max-frequency = <4800000>;
+		rx-sample-delay-ns = <10>;
+	};
+};
+&d1_ssi0 {
+	status = "disabled";
+};
+
+&ssi1 {
+	/* spi flash */
+	status = "disabled";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spi2_default>;
+	num-cs = <2>;
+	spi-flash@0 {
+		compatible = "winbond,w25q128fw",
+					"jedec,spi-nor";
+		reg = <0>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		spi-max-frequency = <4800000>;
+		rx-sample-delay-ns = <10>;
+	};
+	spi-flash@1 {
+		compatible = "winbond,w25q128fw",
+					"jedec,spi-nor";
+		reg = <1>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		spi-max-frequency = <4800000>;
+		rx-sample-delay-ns = <10>;
+	};
+};
+
+&bootspi {
+	/* spi flash */
+	status = "okay";
+	num-cs = <1>;
+	cs-gpios = <&portd 0 GPIO_ACTIVE_LOW>;
+	wp-gpios = <&portd 4 GPIO_ACTIVE_LOW>;
+	spi-flash@0 {
+		compatible = "winbond,w25q128jw",
+					"jedec,spi-nor";
+		reg = <0>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		spi-max-frequency = <4800000>;
+		rx-sample-delay-ns = <10>;
+	};
+};
+
+&d0_mbox0 {
+	status = "okay";
+};
+&d1_mbox0 {
+	status = "okay";
+};
+
+&d0_mbox1 {
+	status = "okay";
+};
+&d1_mbox1 {
+	status = "okay";
+};
+
+&d0_mbox2 {
+	status = "okay";
+};
+&d1_mbox2 {
+	status = "okay";
+};
+
+&d0_mbox3 {
+	status = "okay";
+};
+&d1_mbox3 {
+	status = "okay";
+};
+
+&d0_mbox4 {
+	status = "okay";
+};
+&d1_mbox4 {
+	status = "okay";
+};
+
+&d0_mbox5 {
+	status = "okay";
+};
+&d1_mbox5 {
+	status = "okay";
+};
+
+&d0_mbox6 {
+	status = "okay";
+};
+&d1_mbox6 {
+	status = "okay";
+};
+
+&d0_mbox7 {
+	status = "okay";
+};
+&d1_mbox7 {
+	status = "okay";
+};
+
+&fan_control {
+	status = "okay";
+};
+&d1_fan_control {
+	status = "disabled";
+};
+
+&d0_i2c0 {
+	/* codec es8388 */
+	status = "okay";
+	d0_es8388_0: es8388-0@10 {
+		compatible = "eswin,es8388";
+		reg = <0x10>;
+		#sound-dai-cells = <0>;
+		port {
+			d0_codec0_endpoint: endpoint {
+				system-clock-frequency = <12288000>;
+				remote-endpoint = <&d0_i2s1_endpoint>;
+			};
+		};
+	};
+	d0_es8388_1: es8388-1@11 {
+		compatible = "eswin,es8388";
+		reg = <0x11>;
+		#sound-dai-cells = <0>;
+		port {
+			d0_codec1_endpoint: endpoint {
+				system-clock-frequency = <12288000>;
+				remote-endpoint = <&d0_i2s2_endpoint>;
+			};
+		};
+	};
+};
+&d1_i2c0 {
+	status = "disabled";
+	eeprom@50 {
+		compatible = "atmel,24c1024";
+		reg = <0x50>;
+	};
+};
+
+&d0_i2c1 {
+	/* mpq8785 */
+	status = "okay";
+	mpq8785@10 {
+		compatible = "mps,mpq8785";
+		reg = <0x10>;
+		eswin,regulator_default-microvolt=<1000000>;
+		eswin,regulator_label = "supply vdd1", "npu vdd1", "npu current1", "npu temperature1";
+		label = "npu_vdd";
+		regulators{
+			npu_vcc1:npu_svcc{
+				regulator-name="NPU_SVCC";
+				regulator-min-microvolt=<700000>;
+				regulator-max-microvolt=<1100000>;
+				regulator-min-microamp=<20000000>;
+				regulator-max-microamp=<40000000>;
+				regulator-ov-protection-microvolt=<1100000>;
+				regulator-always-on;
+			};
+		};
+	};
+};
+&d1_i2c1 {
+	status = "disabled";
+};
+
+&d0_i2c2 {
+	/* mipi dsi */
+	status = "disabled";
+};
+&d1_i2c2 {
+	status = "disabled";
+};
+
+&d0_i2c3 {
+	/* mipi csi0/csi1 */
+	status = "okay";
+};
+&d1_i2c3 {
+	status = "disabled";
+};
+
+&d0_i2c4 {
+	/* mipi csi2/csi3 */
+	status = "disabled";
+};
+&d1_i2c4 {
+	status = "disabled";
+};
+
+&d0_i2c5 {
+	/* mipi csi4/csi5 */
+	status = "disabled";
+};
+&d1_i2c5 {
+	status = "disabled";
+};
+
+&d0_i2c6 {
+	/* unused */
+	status = "disabled";
+};
+&d1_i2c6 {
+	status = "disabled";
+};
+
+&d0_i2c7 {
+	/* unused */
+	status = "disabled";
+};
+&d1_i2c7 {
+	status = "disabled";
+};
+
+&d0_i2c8 {
+	/* io extended for mipi csi */
+	status = "okay";
+	tca6416_0: gpio@20 {
+		compatible = "ti,tca6416";
+		reg = <0x20>;
+		gpio-controller; /* IRQ not connected */
+		#gpio-cells = <2>;
+		gpio-line-names = "MIPI_CSI0_PWDN", "MIPI_CSI0_RESET", "MIPI_CSI1_FBC", "MIPI_CSI1_ENB",
+				"MIPI_CSI1_RESET", "MIPI_CSI1_PWDN", "FREX_GP0", "",
+				"MIPI_CSI0_ENB", "MIPI_CSI0_FBC", "FREX_GP2", "MIPI_CSI2_FBC",
+				"MIPI_CSI2_ENB", "FREX_GP1", "MIPI_CSI2_RESET", "MIPI_CSI2_PWDN";
+	};
+	tca6416_1: gpio@21 {
+		compatible = "ti,tca6416";
+		reg = <0x21>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		/* IRQ not connected */
+		gpio-line-names = "MIPI_CSI3_PWDN", "MIPI_CSI3_RESET", "MIPI_CSI3_ENB", "MIPI_CSI3_FBC",
+				"MIPI_CSI4_PWDN", "MIPI_CSI4_RESET", "MIPI_CSI4_ENB", "MIPI_CSI4_FBC",
+				"MIPI_CSI5_FBC", "MIPI_CSI5_ENB", "MIPI_CSI5_RESET", "MIPI_CSI5_PWDN",
+				"", "", "", "";
+	};
+};
+&d1_i2c8 {
+	status = "disabled";
+};
+
+&d0_i2c9 {
+	/* unused */
+	status = "disabled";
+};
+&d1_i2c9 {
+	status = "disabled";
+};
+
+&d0_aon_i2c0 {
+	/* temp sensor & rtc */
+	status = "okay";
+	eswin,syscfg = <&d0_sys_con 0x3C0 16>;
+	pcf_rtc:rtc@51 {
+		compatible = "nxp,pcf8563";
+		reg = <0x51>;
+	};
+};
+&d1_aon_i2c0 {
+	status = "disabled";
+	aon_eeprom@50 {
+		compatible = "atmel,24c1024";
+		reg = <0x50>;
+	};
+};
+
+&d0_aon_i2c1 {
+	/* ina226x4 */
+	status = "okay";
+	i2c-sda-hold-time-ns = <0x40>;
+	ina226@45 {
+		compatible = "ti,ina226";
+		#io-channel-cells = <1>;
+		label = "cpu_vdd";
+		reg = <0x45>;
+		shunt-resistor = <1000>;
+	};
+	ina226@44 {
+			compatible = "ti,ina226";
+			#io-channel-cells = <1>;
+			label = "soc_vdd";
+			reg = <0x44>;
+			shunt-resistor = <1000>;
+		};
+	ina226@41 {
+			compatible = "ti,ina226";
+			#io-channel-cells = <1>;
+			label = "lpddr_vdd";
+			reg = <0x41>;
+			shunt-resistor = <1000>;
+		};
+	ina226@48 {
+			compatible = "ti,ina226";
+			#io-channel-cells = <1>;
+			label = "dc_vdd";
+			reg = <0x48>;
+			shunt-resistor = <1000>;
+		};
+};
+&d1_aon_i2c1 {
+	status = "disabled";
+};
+
+&pwm0 {
+	/* fan */
+	status = "okay";
+};
+&d1_pwm0 {
+	status = "disabled";
+};
+
+&pvt0 {
+	status = "okay";
+};
+&d1_pvt0 {
+	status = "disabled";
+};
+
+&pvt1 {
+	status = "okay";
+};
+&d1_pvt1 {
+	status = "disabled";
+};
+
+&wdt0 {
+	status = "disabled";
+};
+&d1_wdt0 {
+	status = "disabled";
+};
+
+&wdt1 {
+	status = "disabled";
+};
+&d1_wdt1 {
+	status = "disabled";
+};
+
+&wdt2 {
+	status = "disabled";
+};
+&d1_wdt2 {
+	status = "disabled";
+};
+
+&wdt3 {
+	status = "disabled";
+};
+&d1_wdt3 {
+	status = "disabled";
+};
+
+&die0_rtc {
+	status = "okay";
+};
+&d1_rtc {
+	status = "disabled";
+};
+
+&timer0 {
+	status = "okay";
+};
+&d1_timer0 {
+	status = "disabled";
+};
+
+&timer1 {
+	status = "okay";
+};
+&d1_timer1 {
+	status = "disabled";
+};
+
+&timer2 {
+	status = "okay";
+};
+&d1_timer2 {
+	status = "disabled";
+};
+
+&timer3 {
+	status = "okay";
+};
+&d1_timer3 {
+	status = "disabled";
+};
+
+&pinctrl {
+	status = "okay";
+};
+&d1_pinctrl {
+	status = "disabled";
+};
+
+&gpio0 {
+	status = "okay";
+};
+&d1_gpio0 {
+	status = "disabled";
+};
+
+&dev_llc_d0{
+	apply_npu_high_freq;
+};
+&d1_llc_dev{
+	apply_npu_high_freq;
+};
+
+/* die0 GPIO Function Description
+
+	gpio0   : pcie prstn(I)
+	gpio11  : tmp alert(I)
+	gpio13  : tf card detect(I)
+	gpio14  : gphy1 resetn(O)
+	gpio15  : gphy0 resetn(O)
+	gpio18  : ina226 alert(I)
+	gpio20  : lcd touch int(I)
+	gpio21  : head phone plug/unplug detection2(I)
+	gpio27  : black light pwr_en(O)
+	gpio28  : head phone plug/unplug detection1(I)
+	gpio68  : fan pwm(O)
+	gpio69  : fan touch(I)
+	gpio92  : frex gp0(O)
+	gpio93  : frex gp1(O)
+	gpio106 : mipi dsi resetn(O)
+	gpio111 : lcd touch resetn(O)
+*/
+/* die1 GPIO Function Description
+
+	gpio0   : lcd touch int(I)
+	gpio13  : gphy0 resetn(O)
+	gpio14  : gphy1 resetn(O)
+	gpio15  : bt wake host(I)
+	gpio16  : sdio wifi disable(O)
+	gpio18  : sdio wifi wake host(I)
+	gpio19  : pcie prstn(I)
+	gpio20  : head phone plug/unplug detection2(I)
+	gpio21  : mipi dsi resetn(O)
+	gpio27  : tmp alert(I)
+	gpio28  : head phone plug/unplug detection1(I)
+	gpio68  : system led(O)
+	gpio69  : black light pwr_en(O)
+	gpio92  : frex gp0(O)
+	gpio93  : frex gp1(O)
+	gpio106 : host wak bt(O)
+	gpio111 : lcd touch resetn(O)
+*/
\ No newline at end of file
diff --git a/arch/riscv/boot/dts/eswin/eswin-win2030-arch-d2d.dtsi b/arch/riscv/boot/dts/eswin/eswin-win2030-arch-d2d.dtsi
index 828840c3e719..c9afd60d2434 100644
--- a/arch/riscv/boot/dts/eswin/eswin-win2030-arch-d2d.dtsi
+++ b/arch/riscv/boot/dts/eswin/eswin-win2030-arch-d2d.dtsi
@@ -18,868 +18,383 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
-#include <dt-bindings/clock/win2030-clock.h>
-#include "eswin-win2030-platform.dtsi"
+#define CHIPLET_AND_DIE (0x2)
+#include "eswin-win2030-arch.dtsi"
 
-#define UART0_INT	100
-#define UART1_INT	101
-#define UART2_INT	102
 
-/ {
-	#address-cells = <2>;
-	#size-cells = <2>;
-	compatible = "SiFive,FU800-dev", "fu800-dev", "sifive-dev", "eic770x-dev";
-
-	L64: cpus {
-		#address-cells = <1>;
-		#size-cells = <0>;
-		timebase-frequency = <RTCCLK_FREQ>;
-		idle-states {
-			CPU_RET: cpu-retentive {
-				compatible = "riscv,idle-state";
-				riscv,sbi-suspend-param = <0x00000000>;
-				entry-latency-us = <20>;
-				exit-latency-us = <40>;
-				min-residency-us = <80>;
+&L64 {
+	cpu-map {
+	#if (CHIPLET_AND_DIE & 0x2)
+		cluster0 {
+			core0 {
+				cpu = <&cpu_0>;
 			};
-		};
-		cpu-map {
-		#if (CHIPLET_AND_DIE & 0x2)
-			cluster0 {
-				core0 {
-					cpu = <&cpu_0>;
-				};
-				core1 {
-					cpu = <&cpu_1>;
-				};
-				core2 {
-					cpu = <&cpu_2>;
-				};
-				core3 {
-					cpu = <&cpu_3>;
-				};
-			};
-		#endif
-			cluster1 {
-				core0 {
-					cpu = <&cpu_4>;
-				};
-			#ifndef PLATFORM_HAPS
-				core1 {
-					cpu = <&cpu_5>;
-				};
-				core2 {
-					cpu = <&cpu_6>;
-				};
-				core3 {
-					cpu = <&cpu_7>;
-				};
-			#endif
+			core1 {
+				cpu = <&cpu_1>;
 			};
-
-		};
-
-		#if (CHIPLET_AND_DIE & 0x2)
-		cpu_0: cpu@0 {
-			clock-frequency = <0>;
-			compatible = "eswin,eic770x", "riscv";
-			d-cache-block-size = <64>;
-			d-cache-sets = <128>;
-			d-cache-size = <32768>;
-			d-tlb-sets = <1>;
-			d-tlb-size = <32>;
-			device_type = "cpu";
-			hardware-exec-breakpoint-count = <4>;
-			hwpf-distanceBits = <6>;
-			hwpf-hitCacheThrdBits = <5>;
-			hwpf-hitMSHRThrdBits = <4>;
-			hwpf-l2pfPoolSize = <10>;
-			hwpf-nIssQEnt = <6>;
-			hwpf-nPrefetchQueueEntries = <8>;
-			hwpf-nStreams = <16>;
-			hwpf-qFullnessThrdBits = <4>;
-			hwpf-windowBits = <6>;
-			i-cache-block-size = <64>;
-			i-cache-sets = <128>;
-			i-cache-size = <32768>;
-			i-tlb-sets = <1>;
-			i-tlb-size = <32>;
-			mmu-type = "riscv,sv48";
-			next-level-cache = <&L15>;
-			reg = <0x0>;
-			riscv,isa = "rv64imafdch_zicsr_zifencei_zba_zbb";
-			riscv,pmpgranularity = <4096>;
-			riscv,pmpregions = <8>;
-			sifive,buserror = <&L16>;
-			status = "okay";
-			timebase-frequency = <RTCCLK_FREQ>;
-			tlb-split;
-			numa-node-id = <0>;
-			clocks = <&d0_clock WIN2030_CLK_CPU_EXT_SRC_CORE_CLK_0>;
-			operating-points-v2 = <&d0_cpu_opp_table>;
-			cpu-idle-states = <&CPU_RET>;
-			cpu0_intc: interrupt-controller {
-				#interrupt-cells = <1>;
-				compatible = "riscv,cpu-intc";
-				interrupt-controller;
-			};
-			L13: pmu {
-				compatible = "riscv,pmu0", "riscv,pmu";
-				interrupts-extended = <&cpu0_intc 13>;
-			};
-		};
-		cpu_1: cpu@1 {
-			clock-frequency = <0>;
-			compatible = "eswin,eic770x", "riscv";
-			d-cache-block-size = <64>;
-			d-cache-sets = <128>;
-			d-cache-size = <32768>;
-			d-tlb-sets = <1>;
-			d-tlb-size = <32>;
-			device_type = "cpu";
-			hardware-exec-breakpoint-count = <4>;
-			hwpf-distanceBits = <6>;
-			hwpf-hitCacheThrdBits = <5>;
-			hwpf-hitMSHRThrdBits = <4>;
-			hwpf-l2pfPoolSize = <10>;
-			hwpf-nIssQEnt = <6>;
-			hwpf-nPrefetchQueueEntries = <8>;
-			hwpf-nStreams = <16>;
-			hwpf-qFullnessThrdBits = <4>;
-			hwpf-windowBits = <6>;
-			i-cache-block-size = <64>;
-			i-cache-sets = <128>;
-			i-cache-size = <32768>;
-			i-tlb-sets = <1>;
-			i-tlb-size = <32>;
-			mmu-type = "riscv,sv48";
-			next-level-cache = <&L20>;
-			reg = <0x1>;
-			riscv,isa = "rv64imafdch_zicsr_zifencei_zba_zbb";
-			riscv,pmpgranularity = <4096>;
-			riscv,pmpregions = <8>;
-			sifive,buserror = <&L21>;
-			status = "okay";
-			timebase-frequency = <RTCCLK_FREQ>;
-			tlb-split;
-			numa-node-id = <0>;
-			clocks = <&d0_clock WIN2030_CLK_CPU_EXT_SRC_CORE_CLK_1>;
-			operating-points-v2 = <&d0_cpu_opp_table>;
-			cpu-idle-states = <&CPU_RET>;
-			cpu1_intc: interrupt-controller {
-				#interrupt-cells = <1>;
-				compatible = "riscv,cpu-intc";
-				interrupt-controller;
+			core2 {
+				cpu = <&cpu_2>;
 			};
-			L18: pmu {
-				compatible = "riscv,pmu0", "riscv,pmu";
-				interrupts-extended = <&cpu1_intc 13>;
+			core3 {
+				cpu = <&cpu_3>;
 			};
 		};
-		cpu_2: cpu@2 {
-			clock-frequency = <0>;
-			compatible = "eswin,eic770x", "riscv";
-			d-cache-block-size = <64>;
-			d-cache-sets = <128>;
-			d-cache-size = <32768>;
-			d-tlb-sets = <1>;
-			d-tlb-size = <32>;
-			device_type = "cpu";
-			hardware-exec-breakpoint-count = <4>;
-			hwpf-distanceBits = <6>;
-			hwpf-hitCacheThrdBits = <5>;
-			hwpf-hitMSHRThrdBits = <4>;
-			hwpf-l2pfPoolSize = <10>;
-			hwpf-nIssQEnt = <6>;
-			hwpf-nPrefetchQueueEntries = <8>;
-			hwpf-nStreams = <16>;
-			hwpf-qFullnessThrdBits = <4>;
-			hwpf-windowBits = <6>;
-			i-cache-block-size = <64>;
-			i-cache-sets = <128>;
-			i-cache-size = <32768>;
-			i-tlb-sets = <1>;
-			i-tlb-size = <32>;
-			mmu-type = "riscv,sv48";
-			next-level-cache = <&L25>;
-			reg = <0x2>;
-			riscv,isa = "rv64imafdch_zicsr_zifencei_zba_zbb";
-			riscv,pmpgranularity = <4096>;
-			riscv,pmpregions = <8>;
-			sifive,buserror = <&L26>;
-			status = "okay";
-			timebase-frequency = <RTCCLK_FREQ>;
-			tlb-split;
-			numa-node-id = <0>;
-			clocks = <&d0_clock WIN2030_CLK_CPU_EXT_SRC_CORE_CLK_2>;
-			operating-points-v2 = <&d0_cpu_opp_table>;
-			cpu-idle-states = <&CPU_RET>;
-			cpu2_intc: interrupt-controller {
-				#interrupt-cells = <1>;
-				compatible = "riscv,cpu-intc";
-				interrupt-controller;
+	#endif
+		cluster1 {
+			core0 {
+				cpu = <&cpu_4>;
 			};
-			L23: pmu {
-				compatible = "riscv,pmu0", "riscv,pmu";
-				interrupts-extended = <&cpu2_intc 13>;
+		#ifndef PLATFORM_HAPS
+			core1 {
+				cpu = <&cpu_5>;
 			};
-		};
-		cpu_3: cpu@3 {
-			clock-frequency = <0>;
-			compatible = "eswin,eic770x", "riscv";
-			d-cache-block-size = <64>;
-			d-cache-sets = <128>;
-			d-cache-size = <32768>;
-			d-tlb-sets = <1>;
-			d-tlb-size = <32>;
-			device_type = "cpu";
-			hardware-exec-breakpoint-count = <4>;
-			hwpf-distanceBits = <6>;
-			hwpf-hitCacheThrdBits = <5>;
-			hwpf-hitMSHRThrdBits = <4>;
-			hwpf-l2pfPoolSize = <10>;
-			hwpf-nIssQEnt = <6>;
-			hwpf-nPrefetchQueueEntries = <8>;
-			hwpf-nStreams = <16>;
-			hwpf-qFullnessThrdBits = <4>;
-			hwpf-windowBits = <6>;
-			i-cache-block-size = <64>;
-			i-cache-sets = <128>;
-			i-cache-size = <32768>;
-			i-tlb-sets = <1>;
-			i-tlb-size = <32>;
-			mmu-type = "riscv,sv48";
-			next-level-cache = <&L30>;
-			reg = <0x3>;
-			riscv,isa = "rv64imafdch_zicsr_zifencei_zba_zbb";
-			riscv,pmpgranularity = <4096>;
-			riscv,pmpregions = <8>;
-			sifive,buserror = <&L31>;
-			status = "okay";
-			timebase-frequency = <RTCCLK_FREQ>;
-			tlb-split;
-			numa-node-id = <0>;
-			clocks = <&d0_clock WIN2030_CLK_CPU_EXT_SRC_CORE_CLK_3>;
-			operating-points-v2 = <&d0_cpu_opp_table>;
-			cpu-idle-states = <&CPU_RET>;
-			cpu3_intc: interrupt-controller {
-				#interrupt-cells = <1>;
-				compatible = "riscv,cpu-intc";
-				interrupt-controller;
+			core2 {
+				cpu = <&cpu_6>;
 			};
-			L28: pmu {
-				compatible = "riscv,pmu0", "riscv,pmu";
-				interrupts-extended = <&cpu3_intc 13>;
+			core3 {
+				cpu = <&cpu_7>;
 			};
-		};
-		#endif
-
-		cpu_4: cpu@4 {
-			clock-frequency = <0>;
-			compatible = "eswin,eic770x", "riscv";
-			d-cache-block-size = <64>;
-			d-cache-sets = <128>;
-			d-cache-size = <32768>;
-			d-tlb-sets = <1>;
-			d-tlb-size = <32>;
-			device_type = "cpu";
-			hardware-exec-breakpoint-count = <4>;
-			hwpf-distanceBits = <6>;
-			hwpf-hitCacheThrdBits = <5>;
-			hwpf-hitMSHRThrdBits = <4>;
-			hwpf-l2pfPoolSize = <10>;
-			hwpf-nIssQEnt = <6>;
-			hwpf-nPrefetchQueueEntries = <8>;
-			hwpf-nStreams = <16>;
-			hwpf-qFullnessThrdBits = <4>;
-			hwpf-windowBits = <6>;
-			i-cache-block-size = <64>;
-			i-cache-sets = <128>;
-			i-cache-size = <32768>;
-			i-tlb-sets = <1>;
-			i-tlb-size = <32>;
-			mmu-type = "riscv,sv48";
-			next-level-cache = <&D2L2_0>;
-		#if (CHIPLET_AND_DIE == 1)
-		#ifdef PLATFORM_HAPS
-			reg = <0x1>;
-		#else
-			reg = <0x4>;
-		#endif
-		#else
-			reg = <0x4>;
 		#endif
-			riscv,isa = "rv64imafdch_zicsr_zifencei_zba_zbb";
-			riscv,pmpgranularity = <4096>;
-			riscv,pmpregions = <8>;
-			#if (CHIPLET_AND_DIE & 0x2)
-			sifive,buserror = <&L16>;
-			#endif
-			status = "okay";
-			timebase-frequency = <RTCCLK_FREQ>;
-			tlb-split;
-			numa-node-id = <1>;
-			clocks = <&d1_clock WIN2030_CLK_CPU_EXT_SRC_CORE_CLK_0>;
-			operating-points-v2 = <&d1_cpu_opp_table>;
-			cpu-idle-states = <&CPU_RET>;
-			cpu4_intc: interrupt-controller {
-				#interrupt-cells = <1>;
-				compatible = "riscv,cpu-intc";
-				interrupt-controller;
-			};
-
 		};
-	#ifndef PLATFORM_HAPS
-		cpu_5: cpu@5 {
-			clock-frequency = <0>;
-			compatible = "eswin,eic770x", "riscv";
-			d-cache-block-size = <64>;
-			d-cache-sets = <128>;
-			d-cache-size = <32768>;
-			d-tlb-sets = <1>;
-			d-tlb-size = <32>;
-			device_type = "cpu";
-			hardware-exec-breakpoint-count = <4>;
-			hwpf-distanceBits = <6>;
-			hwpf-hitCacheThrdBits = <5>;
-			hwpf-hitMSHRThrdBits = <4>;
-			hwpf-l2pfPoolSize = <10>;
-			hwpf-nIssQEnt = <6>;
-			hwpf-nPrefetchQueueEntries = <8>;
-			hwpf-nStreams = <16>;
-			hwpf-qFullnessThrdBits = <4>;
-			hwpf-windowBits = <6>;
-			i-cache-block-size = <64>;
-			i-cache-sets = <128>;
-			i-cache-size = <32768>;
-			i-tlb-sets = <1>;
-			i-tlb-size = <32>;
-			mmu-type = "riscv,sv48";
-			next-level-cache = <&D2L2_1>;
-			reg = <0x5>;
-			riscv,isa = "rv64imafdch_zicsr_zifencei_zba_zbb";
-			riscv,pmpgranularity = <4096>;
-			riscv,pmpregions = <8>;
-			#if (CHIPLET_AND_DIE & 0x2)
-			sifive,buserror = <&L21>;
-			#endif
-			status = "okay";
-			timebase-frequency = <RTCCLK_FREQ>;
-			tlb-split;
-			numa-node-id = <1>;
-			clocks = <&d1_clock WIN2030_CLK_CPU_EXT_SRC_CORE_CLK_1>;
-			operating-points-v2 = <&d1_cpu_opp_table>;
-			cpu-idle-states = <&CPU_RET>;
-			cpu5_intc: interrupt-controller {
-				#interrupt-cells = <1>;
-				compatible = "riscv,cpu-intc";
-				interrupt-controller;
-			};
-
-		};
-		cpu_6: cpu@6 {
-			clock-frequency = <0>;
-			compatible = "eswin,eic770x", "riscv";
-			d-cache-block-size = <64>;
-			d-cache-sets = <128>;
-			d-cache-size = <32768>;
-			d-tlb-sets = <1>;
-			d-tlb-size = <32>;
-			device_type = "cpu";
-			hardware-exec-breakpoint-count = <4>;
-			hwpf-distanceBits = <6>;
-			hwpf-hitCacheThrdBits = <5>;
-			hwpf-hitMSHRThrdBits = <4>;
-			hwpf-l2pfPoolSize = <10>;
-			hwpf-nIssQEnt = <6>;
-			hwpf-nPrefetchQueueEntries = <8>;
-			hwpf-nStreams = <16>;
-			hwpf-qFullnessThrdBits = <4>;
-			hwpf-windowBits = <6>;
-			i-cache-block-size = <64>;
-			i-cache-sets = <128>;
-			i-cache-size = <32768>;
-			i-tlb-sets = <1>;
-			i-tlb-size = <32>;
-			mmu-type = "riscv,sv48";
-			next-level-cache = <&D2L2_2>;
-			reg = <0x6>;
-			riscv,isa = "rv64imafdch_zicsr_zifencei_zba_zbb";
-			riscv,pmpgranularity = <4096>;
-			riscv,pmpregions = <8>;
-			#if (CHIPLET_AND_DIE & 0x2)
-			sifive,buserror = <&L26>;
-			#endif
-			status = "okay";
-			timebase-frequency = <RTCCLK_FREQ>;
-			tlb-split;
-			numa-node-id = <1>;
-			clocks = <&d1_clock WIN2030_CLK_CPU_EXT_SRC_CORE_CLK_2>;
-			operating-points-v2 = <&d1_cpu_opp_table>;
-			cpu-idle-states = <&CPU_RET>;
-			cpu6_intc: interrupt-controller {
-				#interrupt-cells = <1>;
-				compatible = "riscv,cpu-intc";
-				interrupt-controller;
-			};
-
-		};
-		cpu_7: cpu@7 {
-			clock-frequency = <0>;
-			compatible = "eswin,eic770x", "riscv";
-			d-cache-block-size = <64>;
-			d-cache-sets = <128>;
-			d-cache-size = <32768>;
-			d-tlb-sets = <1>;
-			d-tlb-size = <32>;
-			device_type = "cpu";
-			hardware-exec-breakpoint-count = <4>;
-			hwpf-distanceBits = <6>;
-			hwpf-hitCacheThrdBits = <5>;
-			hwpf-hitMSHRThrdBits = <4>;
-			hwpf-l2pfPoolSize = <10>;
-			hwpf-nIssQEnt = <6>;
-			hwpf-nPrefetchQueueEntries = <8>;
-			hwpf-nStreams = <16>;
-			hwpf-qFullnessThrdBits = <4>;
-			hwpf-windowBits = <6>;
-			i-cache-block-size = <64>;
-			i-cache-sets = <128>;
-			i-cache-size = <32768>;
-			i-tlb-sets = <1>;
-			i-tlb-size = <32>;
-			mmu-type = "riscv,sv48";
-			next-level-cache = <&D2L2_3>;
-			reg = <0x7>;
-			riscv,isa = "rv64imafdch_zicsr_zifencei_zba_zbb";
-			riscv,pmpgranularity = <4096>;
-			riscv,pmpregions = <8>;
-			#if (CHIPLET_AND_DIE & 0x2)
-			sifive,buserror = <&L31>;
-			#endif
-			status = "okay";
-			timebase-frequency = <RTCCLK_FREQ>;
-			tlb-split;
-			numa-node-id = <1>;
-			clocks = <&d1_clock WIN2030_CLK_CPU_EXT_SRC_CORE_CLK_3>;
-			operating-points-v2 = <&d1_cpu_opp_table>;
-			cpu-idle-states = <&CPU_RET>;
-			cpu7_intc: interrupt-controller {
-				#interrupt-cells = <1>;
-				compatible = "riscv,cpu-intc";
-				interrupt-controller;
-			};
 
-		};
-	#endif
 	};
-	//D1MEM: memory@80000000 {
-	//	compatible = "sifive,axi4-mem-port", "sifive,axi4-port", "sifive,mem-port";
-	//	device_type = "memory";
-	//	reg = <0x0 0x80000000 0xf 0x80000000>;
-	//	sifive,port-width-bytes = <32>;
-	//};
 
-	//D2MEM: memory@2000000000 {
-	//	compatible = "sifive,axi4-mem-port", "sifive,axi4-port", "sifive,mem-port";
-	//	device_type = "memory";
-	//	reg = <0x20 0x00000000 0x10 0x00000000>;
-	//	sifive,port-width-bytes = <32>;
-	//};
-
-	SOC: soc {
-		#address-cells = <2>;
-		#size-cells = <2>;
-		compatible = "SiFive,FU800-soc", "fu800-soc", "sifive-soc", "simple-bus";
-		ranges;
-		L40: authentication-controller {
-			compatible = "sifive,authentication0";
-			sifive,auth-types = "fuse";
-		};
-		L51: axi4-sys-port@40000000 {
-			#address-cells = <1>;
-			#size-cells = <1>;
-			compatible = "sifive,axi4-sys-port", "sifive,axi4-port", "sifive,sys-port", "simple-external-bus", "simple-bus";
-			ranges = <0x40000000 0x0 0x40000000 0x40000000>;
-			sifive,port-width-bytes = <16>;
-		};
-		L52: axi4-sys-port@8000000000 {
-			#address-cells = <2>;
-			#size-cells = <2>;
-			compatible = "sifive,axi4-sys-port", "sifive,axi4-port", "sifive,sys-port", "simple-external-bus", "simple-bus";
-			ranges = <0x80 0x0 0x80 0x0 0x180 0x0>;
-			sifive,port-width-bytes = <16>;
-		};
-		L46: basic-bus-blocker@200000 {
-			compatible = "sifive,basic-bus-blocker1";
-			reg = <0x0 0x200000 0x0 0x1000>;
-			reg-names = "control";
-		};
-		L43: basic-bus-blocker@202000 {
-			compatible = "sifive,basic-bus-blocker1";
-			reg = <0x0 0x202000 0x0 0x1000>;
-			reg-names = "control";
-		};
-		L48: basic-bus-blocker@204000 {
-			compatible = "sifive,basic-bus-blocker1";
-			reg = <0x0 0x204000 0x0 0x1000>;
-			reg-names = "control";
-		};
-		L54: burst-bundler@10010000 {
-			compatible = "sifive,burst-bundler0";
-			reg = <0x0 0x10010000 0x0 0x1000>;
-			reg-names = "control";
-		};
+	cpu_4: cpu@4 {
+		clock-frequency = <0>;
+		compatible = "eswin,eic770x", "riscv";
+		d-cache-block-size = <64>;
+		d-cache-sets = <128>;
+		d-cache-size = <32768>;
+		d-tlb-sets = <1>;
+		d-tlb-size = <32>;
+		device_type = "cpu";
+		hardware-exec-breakpoint-count = <4>;
+		hwpf-distanceBits = <6>;
+		hwpf-hitCacheThrdBits = <5>;
+		hwpf-hitMSHRThrdBits = <4>;
+		hwpf-l2pfPoolSize = <10>;
+		hwpf-nIssQEnt = <6>;
+		hwpf-nPrefetchQueueEntries = <8>;
+		hwpf-nStreams = <16>;
+		hwpf-qFullnessThrdBits = <4>;
+		hwpf-windowBits = <6>;
+		i-cache-block-size = <64>;
+		i-cache-sets = <128>;
+		i-cache-size = <32768>;
+		i-tlb-sets = <1>;
+		i-tlb-size = <32>;
+		mmu-type = "riscv,sv48";
+		next-level-cache = <&d1_l2_cache_0>;
+	#if (CHIPLET_AND_DIE == 1)
+	#ifdef PLATFORM_HAPS
+		reg = <0x1>;
+	#else
+		reg = <0x4>;
+	#endif
+	#else
+		reg = <0x4>;
+	#endif
+		riscv,isa = "rv64imafdch_zicsr_zifencei_zba_zbb_sscofpmf";
+		riscv,pmpgranularity = <4096>;
+		riscv,pmpregions = <8>;
 		#if (CHIPLET_AND_DIE & 0x2)
-		L16: bus-error-unit@1700000 {
-			compatible = "sifive,buserror";
-			interrupt-parent = <&plic0>;
-			interrupts = <517>;
-			reg = <0x0 0x1700000 0x0 0x1000>;
-			reg-names = "control";
-		};
-		L21: bus-error-unit@1701000 {
-			compatible = "sifive,buserror";
-			interrupt-parent = <&plic0>;
-			interrupts = <518>;
-			reg = <0x0 0x1701000 0x0 0x1000>;
-			reg-names = "control";
-		};
-		L26: bus-error-unit@1702000 {
-			compatible = "sifive,buserror";
-			interrupt-parent = <&plic0>;
-			interrupts = <519>;
-			reg = <0x0 0x1702000 0x0 0x1000>;
-			reg-names = "control";
-		};
-		L31: bus-error-unit@1703000 {
-			compatible = "sifive,buserror";
-			interrupt-parent = <&plic0>;
-			interrupts = <520>;
-			reg = <0x0 0x1703000 0x0 0x1000>;
-			reg-names = "control";
-		};
+		sifive,buserror = <&d1_bus_err0>;
 		#endif
-
-		d1_bus_err0: bus-error-unit@21700000 {
-			compatible = "sifive,buserror";
-			interrupt-parent = <&plic1>;
-			interrupts = <517>;
-			reg = <0x0 0x21700000 0x0 0x1000>;
-			reg-names = "control";
-		};
-		d1_bus_err1: bus-error-unit@21701000 {
-			compatible = "sifive,buserror";
-			interrupt-parent = <&plic1>;
-			interrupts = <518>;
-			reg = <0x0 0x21701000 0x0 0x1000>;
-			reg-names = "control";
-		};
-		d1_bus_err2: bus-error-unit@21702000 {
-			compatible = "sifive,buserror";
-			interrupt-parent = <&plic1>;
-			interrupts = <519>;
-			reg = <0x0 0x21702000 0x0 0x1000>;
-			reg-names = "control";
-		};
-		d1_bus_err3: bus-error-unit@21703000 {
-			compatible = "sifive,buserror";
-			interrupt-parent = <&plic1>;
-			interrupts = <520>;
-			reg = <0x0 0x21703000 0x0 0x1000>;
-			reg-names = "control";
+		status = "okay";
+		timebase-frequency = <RTCCLK_FREQ>;
+		tlb-split;
+		numa-node-id = <1>;
+		clocks = <&d1_clock WIN2030_CLK_CPU_EXT_SRC_CORE_CLK_0>;
+		operating-points-v2 = <&d1_cpu_opp_table>;
+		cpu-idle-states = <&CPU_RET>;
+		cpu4_intc: interrupt-controller {
+			#interrupt-cells = <1>;
+			compatible = "riscv,cpu-intc";
+			interrupt-controller;
 		};
 
+	};
+#ifndef PLATFORM_HAPS
+	cpu_5: cpu@5 {
+		clock-frequency = <0>;
+		compatible = "eswin,eic770x", "riscv";
+		d-cache-block-size = <64>;
+		d-cache-sets = <128>;
+		d-cache-size = <32768>;
+		d-tlb-sets = <1>;
+		d-tlb-size = <32>;
+		device_type = "cpu";
+		hardware-exec-breakpoint-count = <4>;
+		hwpf-distanceBits = <6>;
+		hwpf-hitCacheThrdBits = <5>;
+		hwpf-hitMSHRThrdBits = <4>;
+		hwpf-l2pfPoolSize = <10>;
+		hwpf-nIssQEnt = <6>;
+		hwpf-nPrefetchQueueEntries = <8>;
+		hwpf-nStreams = <16>;
+		hwpf-qFullnessThrdBits = <4>;
+		hwpf-windowBits = <6>;
+		i-cache-block-size = <64>;
+		i-cache-sets = <128>;
+		i-cache-size = <32768>;
+		i-tlb-sets = <1>;
+		i-tlb-size = <32>;
+		mmu-type = "riscv,sv48";
+		next-level-cache = <&d1_l2_cache_1>;
+		reg = <0x5>;
+		riscv,isa = "rv64imafdch_zicsr_zifencei_zba_zbb_sscofpmf";
+		riscv,pmpgranularity = <4096>;
+		riscv,pmpregions = <8>;
 		#if (CHIPLET_AND_DIE & 0x2)
-		D1CACHE: cache-controller@2010000 {
-			cache-block-size = <64>;
-			cache-level = <3>;
-			cache-sets = <4096>;
-			cache-size = <4194304>;
-			cache-unified;
-			compatible = "sifive,ccache1", "cache", "sifive,fu740-c000-ccache";
-			interrupt-parent = <&plic0>;
-			interrupts = <1>, <3>, <4>, <2>;
-			//next-level-cache = <&L9 &L10 &L11 &D1MEM>;
-			next-level-cache = <&L9 &L10 &L11>;
-			reg = <0x0 0x2010000 0x0 0x4000 0x0 0x8000000 0x0 0x400000>;
-			reg-names = "control", "sideband";
-			sifive,a-mshr-count = <60>;
-			sifive,bank-count = <4>;
-			sifive,ecc-granularity = <8>;
-			sifive,max-master-id = <13>;
-			sifive,perfmon-counters = <6>;
-			numa-node-id = <0>;
-		};
+		sifive,buserror = <&d1_bus_err1>;
 		#endif
-
-		D2CACHE: cache-controller@22010000 {
-			cache-block-size = <64>;
-			cache-level = <3>;
-			cache-sets = <4096>;
-			cache-size = <4194304>;
-			cache-unified;
-			compatible = "sifive,ccache1", "cache", "sifive,fu740-c000-ccache";
-			interrupt-parent = <&plic1>;
-			interrupts = <1>, <3>, <4>, <2>;
-			//next-level-cache = <&L9 &L10 &L11 &D2MEM>;
-			next-level-cache = <&L9 &L10 &L11>;
-			reg = <0x0 0x22010000 0x0 0x4000 0x0 0x8000000 0x0 0x400000>;
-			reg-names = "control", "sideband";
-			sifive,a-mshr-count = <60>;
-			sifive,bank-count = <4>;
-			sifive,ecc-granularity = <8>;
-			sifive,max-master-id = <13>;
-			sifive,perfmon-counters = <6>;
-			numa-node-id = <1>;
-		};
-
-	/*
-		clint0: clint@2000000 {
-			compatible = "riscv,clint0";
-			interrupts-extended = <&cpu0_intc 3 &cpu0_intc 7 &cpu1_intc 3 &cpu1_intc 7 &cpu2_intc 3 &cpu2_intc 7 &cpu3_intc 3 &cpu3_intc 7>;
-			reg = <0x0 0x2000000 0x0 0x10000>;
-			reg-names = "control";
-		};
-
-		clint1: clint@22000000 {
-			compatible = "riscv,clint0";
-			interrupts-extended = <&cpu4_intc 3 &cpu4_intc 7 &cpu5_intc 3 &cpu5_intc 7 &cpu6_intc 3 &cpu6_intc 7 &cpu7_intc 3 &cpu7_intc 7>;
-			reg = <0x0 0x2000000 0x0 0x10000>;
-			reg-names = "control";
+		status = "okay";
+		timebase-frequency = <RTCCLK_FREQ>;
+		tlb-split;
+		numa-node-id = <1>;
+		clocks = <&d1_clock WIN2030_CLK_CPU_EXT_SRC_CORE_CLK_1>;
+		operating-points-v2 = <&d1_cpu_opp_table>;
+		cpu-idle-states = <&CPU_RET>;
+		cpu5_intc: interrupt-controller {
+			#interrupt-cells = <1>;
+			compatible = "riscv,cpu-intc";
+			interrupt-controller;
 		};
-	*/
 
-		L34: debug-controller@0 {
-			compatible = "sifive,debug-100", "riscv,debug-100";
-			debug-attach = "jtag";
-			reg = <0x0 0x0 0x0 0x1000>;
-			reg-names = "control";
-		};
-		L8: error-device@1000 {
-			compatible = "sifive,error0";
-			reg = <0x0 0x1000 0x0 0x3000 0x0 0x5000 0x0 0x13000 0x0 0x19000 0x0 0xe7000 0x0 0x114000 0x0 0xec000 0x0 0x201000 0x0 0x1000 0x0 0x203000 0x0 0x1000 0x0 0x205000 0x0 0x14fb000 0x0 0x1704000 0x0 0x8fc000 0x0 0x2014000 0x0 0x5fec000 0x0 0x8400000 0x0 0x3c00000 0x0 0x10000000 0x0 0x3000 0x0 0x10004000 0x0 0xc000 0x0 0x10011000 0x0 0x1f000 0x0 0x10034000 0x0 0x9fcc000 0x0 0x1a400000 0x0 0x5c00000>;
-		};
-		L9: error-device@10003000 {
-			compatible = "sifive,error0";
-			reg = <0x0 0x10003000 0x0 0x1000>;
-		};
-        /*
-		L49: global-external-interrupts {
-			compatible = "sifive,global-external-interrupts0";
-			interrupt-parent = <&plic0>;
-			interrupts = <5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255 256 257 258 259 260 261 262 263 264 265 266 267 268 269 270 271 272 273 274 275 276 277 278 279 280 281 282 283 284 285 286 287 288 289 290 291 292 293 294 295 296 297 298 299 300 301 302 303 304 305 306 307 308 309 310 311 312 313 314 315 316 317 318 319 320 321 322 323 324 325 326 327 328 329 330 331 332 333 334 335 336 337 338 339 340 341 342 343 344 345 346 347 348 349 350 351 352 353 354 355 356 357 358 359 360 361 362 363 364 365 366 367 368 369 370 371 372 373 374 375 376 377 378 379 380 381 382 383 384 385 386 387 388 389 390 391 392 393 394 395 396 397 398 399 400 401 402 403 404 405 406 407 408 409 410 411 412 413 414 415 416 417 418 419 420 421 422 423 424 425 426 427 428 429 430 431 432 433 434 435 436 437 438 439 440 441 442 443 444 445 446 447 448 449 450 451 452 453 454 455 456 457 458 459 460 461 462 463 464 465 466 467 468 469 470 471 472 473 474 475 476 477 478 479 480 481 482 483 484 485 486 487 488 489 490 491 492 493 494 495 496 497 498 499 500 501 502 503 504 505 506 507 508 509 510 511 512 513 514 515>;
-		};
-        */
+	};
+	cpu_6: cpu@6 {
+		clock-frequency = <0>;
+		compatible = "eswin,eic770x", "riscv";
+		d-cache-block-size = <64>;
+		d-cache-sets = <128>;
+		d-cache-size = <32768>;
+		d-tlb-sets = <1>;
+		d-tlb-size = <32>;
+		device_type = "cpu";
+		hardware-exec-breakpoint-count = <4>;
+		hwpf-distanceBits = <6>;
+		hwpf-hitCacheThrdBits = <5>;
+		hwpf-hitMSHRThrdBits = <4>;
+		hwpf-l2pfPoolSize = <10>;
+		hwpf-nIssQEnt = <6>;
+		hwpf-nPrefetchQueueEntries = <8>;
+		hwpf-nStreams = <16>;
+		hwpf-qFullnessThrdBits = <4>;
+		hwpf-windowBits = <6>;
+		i-cache-block-size = <64>;
+		i-cache-sets = <128>;
+		i-cache-size = <32768>;
+		i-tlb-sets = <1>;
+		i-tlb-size = <32>;
+		mmu-type = "riscv,sv48";
+		next-level-cache = <&d1_l2_cache_2>;
+		reg = <0x6>;
+		riscv,isa = "rv64imafdch_zicsr_zifencei_zba_zbb_sscofpmf";
+		riscv,pmpgranularity = <4096>;
+		riscv,pmpregions = <8>;
 		#if (CHIPLET_AND_DIE & 0x2)
-		plic0: interrupt-controller@c000000 {
+		sifive,buserror = <&d1_bus_err2>;
+		#endif
+		status = "okay";
+		timebase-frequency = <RTCCLK_FREQ>;
+		tlb-split;
+		numa-node-id = <1>;
+		clocks = <&d1_clock WIN2030_CLK_CPU_EXT_SRC_CORE_CLK_2>;
+		operating-points-v2 = <&d1_cpu_opp_table>;
+		cpu-idle-states = <&CPU_RET>;
+		cpu6_intc: interrupt-controller {
 			#interrupt-cells = <1>;
-			compatible = "sifive,plic-1.0.0";
+			compatible = "riscv,cpu-intc";
 			interrupt-controller;
-			interrupts-extended = <
-				&cpu0_intc 0xffffffff &cpu0_intc 9
-				&cpu1_intc 0xffffffff &cpu1_intc 9
-				&cpu2_intc 0xffffffff &cpu2_intc 9
-				&cpu3_intc 0xffffffff &cpu3_intc 9>;
-			reg = <0x0 0xc000000 0x0 0x4000000>;
-			reg-names = "control";
-			riscv,max-priority = <7>;
-			riscv,ndev = <520>;
-			numa-node-id = <0>;
 		};
-		#endif
 
-		plic1: interrupt-controller@2c000000 {
+	};
+	cpu_7: cpu@7 {
+		clock-frequency = <0>;
+		compatible = "eswin,eic770x", "riscv";
+		d-cache-block-size = <64>;
+		d-cache-sets = <128>;
+		d-cache-size = <32768>;
+		d-tlb-sets = <1>;
+		d-tlb-size = <32>;
+		device_type = "cpu";
+		hardware-exec-breakpoint-count = <4>;
+		hwpf-distanceBits = <6>;
+		hwpf-hitCacheThrdBits = <5>;
+		hwpf-hitMSHRThrdBits = <4>;
+		hwpf-l2pfPoolSize = <10>;
+		hwpf-nIssQEnt = <6>;
+		hwpf-nPrefetchQueueEntries = <8>;
+		hwpf-nStreams = <16>;
+		hwpf-qFullnessThrdBits = <4>;
+		hwpf-windowBits = <6>;
+		i-cache-block-size = <64>;
+		i-cache-sets = <128>;
+		i-cache-size = <32768>;
+		i-tlb-sets = <1>;
+		i-tlb-size = <32>;
+		mmu-type = "riscv,sv48";
+		next-level-cache = <&d1_l2_cache_3>;
+		reg = <0x7>;
+		riscv,isa = "rv64imafdch_zicsr_zifencei_zba_zbb_sscofpmf";
+		riscv,pmpgranularity = <4096>;
+		riscv,pmpregions = <8>;
+		#if (CHIPLET_AND_DIE & 0x2)
+		sifive,buserror = <&d1_bus_err3>;
+		#endif
+		status = "okay";
+		timebase-frequency = <RTCCLK_FREQ>;
+		tlb-split;
+		numa-node-id = <1>;
+		clocks = <&d1_clock WIN2030_CLK_CPU_EXT_SRC_CORE_CLK_3>;
+		operating-points-v2 = <&d1_cpu_opp_table>;
+		cpu-idle-states = <&CPU_RET>;
+		cpu7_intc: interrupt-controller {
 			#interrupt-cells = <1>;
-			compatible = "sifive,plic-1.0.0";
+			compatible = "riscv,cpu-intc";
 			interrupt-controller;
-			interrupts-extended = <
-				&cpu4_intc 0xffffffff &cpu4_intc 9
-			#ifndef PLATFORM_HAPS
-				&cpu5_intc 0xffffffff &cpu5_intc 9
-				&cpu6_intc 0xffffffff &cpu6_intc 9
-				&cpu7_intc 0xffffffff &cpu7_intc 9
-			#endif
-				>;
-			reg = <0x0 0x2c000000 0x0 0x4000000>;
-			reg-names = "control";
-			riscv,max-priority = <7>;
-			riscv,ndev = <520>;
-			numa-node-id = <1>;
 		};
 
-		#if (CHIPLET_AND_DIE & 0x2)
-		L53: order-obliterator@10030000 {
-			compatible = "sifive,order-obliterator0";
-			interrupt-parent = <&plic0>;
-			interrupts = <516>;
-			reg = <0x0 0x10030000 0x0 0x4000>;
-			reg-names = "control";
-		};
-		L15: pl2@104000 {
-			cache-block-size = <64>;
-			cache-level = <2>;
-			cache-sets = <512>;
-			cache-size = <262144>;
-			cache-unified;
-			compatible = "sifive,pL2Cache0", "cache";
-			next-level-cache = <&D1CACHE>;
-			reg = <0x0 0x104000 0x0 0x4000>;
-			reg-names = "control";
-			sifive,ecc-granularity = <16>;
-			sifive,perfmon-counters = <6>;
-		};
-		L20: pl2@108000 {
-			cache-block-size = <64>;
-			cache-level = <2>;
-			cache-sets = <512>;
-			cache-size = <262144>;
-			cache-unified;
-			compatible = "sifive,pL2Cache0", "cache";
-			next-level-cache = <&D1CACHE>;
-			reg = <0x0 0x108000 0x0 0x4000>;
-			reg-names = "control";
-			sifive,ecc-granularity = <16>;
-			sifive,perfmon-counters = <6>;
-		};
-		L25: pl2@10c000 {
-			cache-block-size = <64>;
-			cache-level = <2>;
-			cache-sets = <512>;
-			cache-size = <262144>;
-			cache-unified;
-			compatible = "sifive,pL2Cache0", "cache";
-			next-level-cache = <&D1CACHE>;
-			reg = <0x0 0x10c000 0x0 0x4000>;
-			reg-names = "control";
-			sifive,ecc-granularity = <16>;
-			sifive,perfmon-counters = <6>;
-		};
-		L30: pl2@110000 {
-			cache-block-size = <64>;
-			cache-level = <2>;
-			cache-sets = <512>;
-			cache-size = <262144>;
-			cache-unified;
-			compatible = "sifive,pL2Cache0", "cache";
-			next-level-cache = <&D1CACHE>;
-			reg = <0x0 0x110000 0x0 0x4000>;
-			reg-names = "control";
-			sifive,ecc-granularity = <16>;
-			sifive,perfmon-counters = <6>;
-		};
-		#endif
+	};
+#endif
+};
 
-		D2L2_0: pl2@20104000 {
-			cache-block-size = <64>;
-			cache-level = <2>;
-			cache-sets = <512>;
-			cache-size = <262144>;
-			cache-unified;
-			compatible = "sifive,pL2Cache0", "cache";
-			next-level-cache = <&D2CACHE>;
-			reg = <0x0 0x20104000 0x0 0x4000>;
-			reg-names = "control";
-			sifive,ecc-granularity = <16>;
-			sifive,perfmon-counters = <6>;
-		};
-		D2L2_1: pl2@20108000 {
-			cache-block-size = <64>;
-			cache-level = <2>;
-			cache-sets = <512>;
-			cache-size = <262144>;
-			cache-unified;
-			compatible = "sifive,pL2Cache0", "cache";
-			next-level-cache = <&D2CACHE>;
-			reg = <0x0 0x20108000 0x0 0x4000>;
-			reg-names = "control";
-			sifive,ecc-granularity = <16>;
-			sifive,perfmon-counters = <6>;
-		};
-		D2L2_2: pl2@2010c000 {
-			cache-block-size = <64>;
-			cache-level = <2>;
-			cache-sets = <512>;
-			cache-size = <262144>;
-			cache-unified;
-			compatible = "sifive,pL2Cache0", "cache";
-			next-level-cache = <&D2CACHE>;
-			reg = <0x0 0x2010c000 0x0 0x4000>;
-			reg-names = "control";
-			sifive,ecc-granularity = <16>;
-			sifive,perfmon-counters = <6>;
-		};
-		D2L2_3: pl2@20110000 {
-			cache-block-size = <64>;
-			cache-level = <2>;
-			cache-sets = <512>;
-			cache-size = <262144>;
-			cache-unified;
-			compatible = "sifive,pL2Cache0", "cache";
-			next-level-cache = <&D2CACHE>;
-			reg = <0x0 0x20110000 0x0 0x4000>;
-			reg-names = "control";
-			sifive,ecc-granularity = <16>;
-			sifive,perfmon-counters = <6>;
-		};
+&SOC {
+	#address-cells = <2>;
+	#size-cells = <2>;
+	compatible = "SiFive,FU800-soc", "fu800-soc", "sifive-soc", "simple-bus";
+	ranges;
+	PMU1: pmu@1 {
+		riscv,raw-event-to-mhpmcounters = <0x0 0x0  0xffffffff 0xff          0x1f8
+											0x0 0x1  0xffffffff 0xfff800ff    0x1f8
+											0x0 0x2  0xffffffff 0xffffc0ff    0x1f8>;
+		riscv,event-to-mhpmcounters = <0x01 0x01 0x01 0x02 0x02 0x02 0x4 0x6 0x1f8 0x10009 0x10009 0x1f8 0x10019 0x10019 0x1f8 0x10021 0x10021 0x1f8>;
+		riscv,event-to-mhpmevent = <0x4 0x0 0x202 0x5 0x0 0x4000 0x6 0x0 0x2001 0x10009 0x0 0x102 0x10019 0x0 0x1002 0x10021 0x0 0x802>;
+		compatible = "riscv,pmu0", "riscv,pmu";
+		interrupts-extended = <&cpu4_intc 13 &cpu5_intc 13 &cpu6_intc 13 &cpu7_intc 13>;
+	};
 
-		L10: rom@1a000000 {
-			compatible = "ucbbar,cacheable-zero0";
-			reg = <0x0 0x1a000000 0x0 0x400000>;
-		};
-		L11: rom@3a000000 {
-			compatible = "ucbbar,cacheable-zero0";
-			reg = <0x0 0x3a000000 0x0 0x400000>;
-		};
-		L6: subsystem_pbus_clock {
-			#clock-cells = <0>;
-			clock-frequency = <10000000>;
-			clock-output-names = "subsystem_pbus_clock";
-			compatible = "fixed-clock";
-		};
-		L61: teststatus@4000 {
-			compatible = "sifive,test0";
-			reg = <0x0 0x4000 0x0 0x1000>;
-			reg-names = "control";
-		};
-		L45: tl-address-adjuster@20000000 {
-			#address-cells = <1>;
-			#size-cells = <1>;
-			compatible = "sifive,tl-inter-sys-port", "sifive,tl-port", "simple-external-bus", "simple-bus";
-			ranges = <0x20000000 0x0 0x20000000 0x1a000000 0x3a400000 0x0 0x3a400000 0x5c00000>;
-			sifive,port-width-bytes = <8>;
-		};
-		L42: tl-inter-mem-master-port@80000000 {
-			#address-cells = <2>;
-			#size-cells = <2>;
-			compatible = "sifive,tl-inter-mem-master-port", "sifive,tl-port", "sifive,inter-mem-master-port", "simple-external-bus", "simple-bus";
-			ranges = <0x0 0x80000000 0x0 0x80000000 0x7f 0x80000000>;
-			sifive,port-width-bytes = <32>;
-		};
-		L55: trace-encoder-0@100000 {
-			compatible = "sifive,trace0";
-			reg = <0x0 0x100000 0x0 0x1000>;
-			reg-names = "control";
-		};
-		L56: trace-encoder-1@101000 {
-			compatible = "sifive,trace0";
-			reg = <0x0 0x101000 0x0 0x1000>;
-			reg-names = "control";
-		};
-		L57: trace-encoder-2@102000 {
-			compatible = "sifive,trace0";
-			reg = <0x0 0x102000 0x0 0x1000>;
-			reg-names = "control";
-		};
-		L58: trace-encoder-3@103000 {
-			compatible = "sifive,trace0";
-			reg = <0x0 0x103000 0x0 0x1000>;
-			reg-names = "control";
-		};
-		L59: trace-funnel-0@18000 {
-			compatible = "sifive,trace0";
-			reg = <0x0 0x18000 0x0 0x1000>;
-			reg-names = "control";
-		};
+	d1_bus_err0: bus-error-unit@21700000 {
+		compatible = "sifive,buserror";
+		interrupt-parent = <&plic1>;
+		interrupts = <517>;
+		reg = <0x0 0x21700000 0x0 0x1000>;
+		reg-names = "control";
+	};
+	d1_bus_err1: bus-error-unit@21701000 {
+		compatible = "sifive,buserror";
+		interrupt-parent = <&plic1>;
+		interrupts = <518>;
+		reg = <0x0 0x21701000 0x0 0x1000>;
+		reg-names = "control";
+	};
+	d1_bus_err2: bus-error-unit@21702000 {
+		compatible = "sifive,buserror";
+		interrupt-parent = <&plic1>;
+		interrupts = <519>;
+		reg = <0x0 0x21702000 0x0 0x1000>;
+		reg-names = "control";
+	};
+	d1_bus_err3: bus-error-unit@21703000 {
+		compatible = "sifive,buserror";
+		interrupt-parent = <&plic1>;
+		interrupts = <520>;
+		reg = <0x0 0x21703000 0x0 0x1000>;
+		reg-names = "control";
+	};
+
+	D1CACHE: cache-controller@22010000 {
+		cache-block-size = <64>;
+		cache-level = <3>;
+		cache-sets = <4096>;
+		cache-size = <4194304>;
+		cache-unified;
+		compatible = "sifive,ccache1", "cache", "sifive,fu740-c000-ccache";
+		interrupt-parent = <&plic1>;
+		interrupts = <1>, <3>, <4>, <2>;
+		//next-level-cache = <&L9 &L10 &L11 &D2MEM>;
+		next-level-cache = <&L9 &L10 &L11>;
+		reg = <0x0 0x22010000 0x0 0x4000 0x0 0x8000000 0x0 0x400000>;
+		reg-names = "control", "sideband";
+		sifive,a-mshr-count = <60>;
+		sifive,bank-count = <4>;
+		sifive,ecc-granularity = <8>;
+		sifive,max-master-id = <13>;
+		sifive,perfmon-counters = <6>;
+		numa-node-id = <1>;
+	};
+
+
+	plic1: interrupt-controller@2c000000 {
+		#interrupt-cells = <1>;
+		compatible = "sifive,plic-1.0.0";
+		interrupt-controller;
+		interrupts-extended = <
+			&cpu4_intc 0xffffffff &cpu4_intc 9
+		#ifndef PLATFORM_HAPS
+			&cpu5_intc 0xffffffff &cpu5_intc 9
+			&cpu6_intc 0xffffffff &cpu6_intc 9
+			&cpu7_intc 0xffffffff &cpu7_intc 9
+		#endif
+			>;
+		reg = <0x0 0x2c000000 0x0 0x4000000>;
+		reg-names = "control";
+		riscv,max-priority = <7>;
+		riscv,ndev = <520>;
+		numa-node-id = <1>;
+	};
+
+	d1_l2_cache_0: pl2@20104000 {
+		cache-block-size = <64>;
+		cache-level = <2>;
+		cache-sets = <512>;
+		cache-size = <262144>;
+		cache-unified;
+		compatible = "sifive,pL2Cache0", "cache";
+		next-level-cache = <&D1CACHE>;
+		reg = <0x0 0x20104000 0x0 0x4000>;
+		reg-names = "control";
+		sifive,ecc-granularity = <16>;
+		sifive,perfmon-counters = <6>;
+	};
+	d1_l2_cache_1: pl2@20108000 {
+		cache-block-size = <64>;
+		cache-level = <2>;
+		cache-sets = <512>;
+		cache-size = <262144>;
+		cache-unified;
+		compatible = "sifive,pL2Cache0", "cache";
+		next-level-cache = <&D1CACHE>;
+		reg = <0x0 0x20108000 0x0 0x4000>;
+		reg-names = "control";
+		sifive,ecc-granularity = <16>;
+		sifive,perfmon-counters = <6>;
+	};
+	d1_l2_cache_2: pl2@2010c000 {
+		cache-block-size = <64>;
+		cache-level = <2>;
+		cache-sets = <512>;
+		cache-size = <262144>;
+		cache-unified;
+		compatible = "sifive,pL2Cache0", "cache";
+		next-level-cache = <&D1CACHE>;
+		reg = <0x0 0x2010c000 0x0 0x4000>;
+		reg-names = "control";
+		sifive,ecc-granularity = <16>;
+		sifive,perfmon-counters = <6>;
+	};
+	d1_l2_cache_3: pl2@20110000 {
+		cache-block-size = <64>;
+		cache-level = <2>;
+		cache-sets = <512>;
+		cache-size = <262144>;
+		cache-unified;
+		compatible = "sifive,pL2Cache0", "cache";
+		next-level-cache = <&D1CACHE>;
+		reg = <0x0 0x20110000 0x0 0x4000>;
+		reg-names = "control";
+		sifive,ecc-granularity = <16>;
+		sifive,perfmon-counters = <6>;
 	};
 };
+
diff --git a/arch/riscv/boot/dts/eswin/eswin-win2030-arch.dtsi b/arch/riscv/boot/dts/eswin/eswin-win2030-arch.dtsi
index bde485b4abfa..a1c618fabcea 100644
--- a/arch/riscv/boot/dts/eswin/eswin-win2030-arch.dtsi
+++ b/arch/riscv/boot/dts/eswin/eswin-win2030-arch.dtsi
@@ -48,7 +48,7 @@ CPU_RET: cpu-retentive {
 				min-residency-us = <80>;
 			};
 		};
-		L17: cpu@0 {
+		cpu_0: cpu@0 {
 			clock-frequency = <0>;
 			compatible = "eswin,eic770x", "riscv";
 			d-cache-block-size = <64>;
@@ -73,12 +73,12 @@ L17: cpu@0 {
 			i-tlb-sets = <1>;
 			i-tlb-size = <32>;
 			mmu-type = "riscv,sv48";
-			next-level-cache = <&L15>;
+			next-level-cache = <&d0_l2_cache_0>;
 			reg = <0x0>;
 			riscv,isa = "rv64imafdch_zicsr_zifencei_zba_zbb_sscofpmf";
 			riscv,pmpgranularity = <4096>;
 			riscv,pmpregions = <8>;
-			sifive,buserror = <&L16>;
+			sifive,buserror = <&d0_bus_err0>;
 			status = "okay";
 			timebase-frequency = <RTCCLK_FREQ>;
 			tlb-split;
@@ -87,13 +87,13 @@ L17: cpu@0 {
 			#cooling-cells = <2>;
 			dynamic-power-coefficient = <324>;
 			cpu-idle-states = <&CPU_RET>;
-			L14: interrupt-controller {
+			cpu0_intc: interrupt-controller {
 				#interrupt-cells = <1>;
 				compatible = "riscv,cpu-intc";
 				interrupt-controller;
 			};
 		};
-		L22: cpu@1 {
+		cpu_1: cpu@1 {
 			clock-frequency = <0>;
 			compatible = "eswin,eic770x", "riscv";
 			d-cache-block-size = <64>;
@@ -118,12 +118,12 @@ L22: cpu@1 {
 			i-tlb-sets = <1>;
 			i-tlb-size = <32>;
 			mmu-type = "riscv,sv48";
-			next-level-cache = <&L20>;
+			next-level-cache = <&d0_l2_cache_1>;
 			reg = <0x1>;
 			riscv,isa = "rv64imafdch_zicsr_zifencei_zba_zbb_sscofpmf";
 			riscv,pmpgranularity = <4096>;
 			riscv,pmpregions = <8>;
-			sifive,buserror = <&L21>;
+			sifive,buserror = <&d0_bus_err1>;
 			status = "okay";
 			timebase-frequency = <RTCCLK_FREQ>;
 			tlb-split;
@@ -132,13 +132,13 @@ L22: cpu@1 {
 			#cooling-cells = <2>;
 			dynamic-power-coefficient = <324>;
 			cpu-idle-states = <&CPU_RET>;
-			L19: interrupt-controller {
+			cpu1_intc: interrupt-controller {
 				#interrupt-cells = <1>;
 				compatible = "riscv,cpu-intc";
 				interrupt-controller;
 			};
 		};
-		L27: cpu@2 {
+		cpu_2: cpu@2 {
 			clock-frequency = <0>;
 			compatible = "eswin,eic770x", "riscv";
 			d-cache-block-size = <64>;
@@ -163,12 +163,12 @@ L27: cpu@2 {
 			i-tlb-sets = <1>;
 			i-tlb-size = <32>;
 			mmu-type = "riscv,sv48";
-			next-level-cache = <&L25>;
+			next-level-cache = <&d0_l2_cache_2>;
 			reg = <0x2>;
 			riscv,isa = "rv64imafdch_zicsr_zifencei_zba_zbb_sscofpmf";
 			riscv,pmpgranularity = <4096>;
 			riscv,pmpregions = <8>;
-			sifive,buserror = <&L26>;
+			sifive,buserror = <&d0_bus_err2>;
 			status = "okay";
 			timebase-frequency = <RTCCLK_FREQ>;
 			tlb-split;
@@ -177,13 +177,13 @@ L27: cpu@2 {
 			#cooling-cells = <2>;
 			dynamic-power-coefficient = <324>;
 			cpu-idle-states = <&CPU_RET>;
-			L24: interrupt-controller {
+			cpu2_intc: interrupt-controller {
 				#interrupt-cells = <1>;
 				compatible = "riscv,cpu-intc";
 				interrupt-controller;
 			};
 		};
-		L32: cpu@3 {
+		cpu_3: cpu@3 {
 			clock-frequency = <0>;
 			compatible = "eswin,eic770x", "riscv";
 			d-cache-block-size = <64>;
@@ -208,12 +208,12 @@ L32: cpu@3 {
 			i-tlb-sets = <1>;
 			i-tlb-size = <32>;
 			mmu-type = "riscv,sv48";
-			next-level-cache = <&L30>;
+			next-level-cache = <&d0_l2_cache_3>;
 			reg = <0x3>;
 			riscv,isa = "rv64imafdch_zicsr_zifencei_zba_zbb_sscofpmf";
 			riscv,pmpgranularity = <4096>;
 			riscv,pmpregions = <8>;
-			sifive,buserror = <&L31>;
+			sifive,buserror = <&d0_bus_err3>;
 			status = "okay";
 			timebase-frequency = <RTCCLK_FREQ>;
 			tlb-split;
@@ -222,7 +222,7 @@ L32: cpu@3 {
 			#cooling-cells = <2>;
 			dynamic-power-coefficient = <324>;
 			cpu-idle-states = <&CPU_RET>;
-			L29: interrupt-controller {
+			cpu3_intc: interrupt-controller {
 				#interrupt-cells = <1>;
 				compatible = "riscv,cpu-intc";
 				interrupt-controller;
@@ -240,14 +240,14 @@ SOC: soc {
 		#size-cells = <2>;
 		compatible = "SiFive,FU800-soc", "fu800-soc", "sifive-soc", "simple-bus";
 		ranges;
-		pmu {
+		PMU0: pmu@0 {
 			riscv,raw-event-to-mhpmcounters = <0x0 0x0  0xffffffff 0xff          0x1f8
 											   0x0 0x1  0xffffffff 0xfff800ff    0x1f8
 											   0x0 0x2  0xffffffff 0xffffc0ff    0x1f8>;
 			riscv,event-to-mhpmcounters = <0x01 0x01 0x01 0x02 0x02 0x02 0x4 0x6 0x1f8 0x10009 0x10009 0x1f8 0x10019 0x10019 0x1f8 0x10021 0x10021 0x1f8>;
 			riscv,event-to-mhpmevent = <0x4 0x0 0x202 0x5 0x0 0x4000 0x6 0x0 0x2001 0x10009 0x0 0x102 0x10019 0x0 0x1002 0x10021 0x0 0x802>;
 			compatible = "riscv,pmu0", "riscv,pmu";
-			interrupts-extended = <&L14 13 &L19 13 &L24 13 &L29 13>;
+			interrupts-extended = <&cpu0_intc 13 &cpu1_intc 13 &cpu2_intc 13 &cpu3_intc 13>;
 		};
 		L40: authentication-controller {
 			compatible = "sifive,authentication0";
@@ -287,35 +287,35 @@ L54: burst-bundler@10010000 {
 			reg = <0x0 0x10010000 0x0 0x1000>;
 			reg-names = "control";
 		};
-		L16: bus-error-unit@hart0 {
+		d0_bus_err0: bus-error-unit@hart0 {
 			compatible = "sifive,buserror";
 			interrupt-parent = <&plic0>;
 			interrupts = <517>;
 			reg = <0x0 0x1700000 0x0 0x1000>;
 			reg-names = "control";
 		};
-		L21: bus-error-unit@hart1 {
+		d0_bus_err1: bus-error-unit@hart1 {
 			compatible = "sifive,buserror";
 			interrupt-parent = <&plic0>;
 			interrupts = <518>;
 			reg = <0x0 0x1701000 0x0 0x1000>;
 			reg-names = "control";
 		};
-		L26: bus-error-unit@hart2 {
+		d0_bus_err2: bus-error-unit@hart2 {
 			compatible = "sifive,buserror";
 			interrupt-parent = <&plic0>;
 			interrupts = <519>;
 			reg = <0x0 0x1702000 0x0 0x1000>;
 			reg-names = "control";
 		};
-		L31: bus-error-unit@hart3 {
+		d0_bus_err3: bus-error-unit@hart3 {
 			compatible = "sifive,buserror";
 			interrupt-parent = <&plic0>;
 			interrupts = <520>;
 			reg = <0x0 0x1703000 0x0 0x1000>;
 			reg-names = "control";
 		};
-		L7: cache-controller@2010000 {
+		D0CACHE: cache-controller@2010000 {
 			cache-block-size = <64>;
 			cache-level = <3>;
 			cache-sets = <4096>;
@@ -353,14 +353,15 @@ plic0: interrupt-controller@c000000 {
 			compatible = "sifive,plic-1.0.0";
 			interrupt-controller;
 			interrupts-extended = <
-				&L14 0xffffffff &L14 9
-				&L19 0xffffffff &L19 9
-				&L24 0xffffffff &L24 9
-				&L29 0xffffffff &L29 9>;
+				&cpu0_intc 0xffffffff &cpu0_intc 9
+				&cpu1_intc 0xffffffff &cpu1_intc 9
+				&cpu2_intc 0xffffffff &cpu2_intc 9
+				&cpu3_intc 0xffffffff &cpu3_intc 9>;
 			reg = <0x0 0xc000000 0x0 0x4000000>;
 			reg-names = "control";
 			riscv,max-priority = <7>;
 			riscv,ndev = <520>;
+			numa-node-id = <0>;
 		};
 		L53: order-obliterator@10030000 {
 			compatible = "sifive,order-obliterator0";
@@ -369,53 +370,53 @@ L53: order-obliterator@10030000 {
 			reg = <0x0 0x10030000 0x0 0x4000>;
 			reg-names = "control";
 		};
-		L15: pl2@104000 {
+		d0_l2_cache_0: pl2@104000 {
 			cache-block-size = <64>;
 			cache-level = <2>;
 			cache-sets = <512>;
 			cache-size = <262144>;
 			cache-unified;
 			compatible = "sifive,pL2Cache0", "cache";
-			next-level-cache = <&L7>;
+			next-level-cache = <&D0CACHE>;
 			reg = <0x0 0x104000 0x0 0x4000>;
 			reg-names = "control";
 			sifive,ecc-granularity = <16>;
 			sifive,perfmon-counters = <6>;
 		};
-		L20: pl2@108000 {
+		d0_l2_cache_1: pl2@108000 {
 			cache-block-size = <64>;
 			cache-level = <2>;
 			cache-sets = <512>;
 			cache-size = <262144>;
 			cache-unified;
 			compatible = "sifive,pL2Cache0", "cache";
-			next-level-cache = <&L7>;
+			next-level-cache = <&D0CACHE>;
 			reg = <0x0 0x108000 0x0 0x4000>;
 			reg-names = "control";
 			sifive,ecc-granularity = <16>;
 			sifive,perfmon-counters = <6>;
 		};
-		L25: pl2@10c000 {
+		d0_l2_cache_2: pl2@10c000 {
 			cache-block-size = <64>;
 			cache-level = <2>;
 			cache-sets = <512>;
 			cache-size = <262144>;
 			cache-unified;
 			compatible = "sifive,pL2Cache0", "cache";
-			next-level-cache = <&L7>;
+			next-level-cache = <&D0CACHE>;
 			reg = <0x0 0x10c000 0x0 0x4000>;
 			reg-names = "control";
 			sifive,ecc-granularity = <16>;
 			sifive,perfmon-counters = <6>;
 		};
-		L30: pl2@110000 {
+		d0_l2_cache_3: pl2@110000 {
 			cache-block-size = <64>;
 			cache-level = <2>;
 			cache-sets = <512>;
 			cache-size = <262144>;
 			cache-unified;
 			compatible = "sifive,pL2Cache0", "cache";
-			next-level-cache = <&L7>;
+			next-level-cache = <&D0CACHE>;
 			reg = <0x0 0x110000 0x0 0x4000>;
 			reg-names = "control";
 			sifive,ecc-granularity = <16>;
diff --git a/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi b/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi
index 9de7c694998a..2a99656c01b4 100644
--- a/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi
+++ b/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi
@@ -18,11 +18,8 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
-#if (!(CHIPLET_AND_DIE & 0x2))
 #include "eswin-win2030-arch.dtsi"
-#else
-#include "eswin-win2030-arch-d2d.dtsi"
-#endif
+
 
 #include <dt-bindings/memory/eswin-win2030-sid.h>
 #include <dt-bindings/mailbox/eswin-mailbox.h>
diff --git a/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi b/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi
index d271b6843621..0a70ba1aad61 100644
--- a/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi
+++ b/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi
@@ -18,11 +18,8 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
-#include "eswin-win2030-platform.dtsi"
-
 #include "eswin-win2030-arch-d2d.dtsi"
 
-
 #include <dt-bindings/memory/eswin-win2030-sid.h>
 #include <dt-bindings/mailbox/eswin-mailbox.h>
 #include <dt-bindings/reset/eswin,win2030-syscrg.h>
@@ -30,6 +27,7 @@
 #include <dt-bindings/i2c/i2c.h>
 #include <dt-bindings/interconnect/eswin,win2030.h>
 #include <dt-bindings/thermal/thermal.h>
+#include <dt-bindings/power/eswin,eic770x-pmu.h>
 
 / {
 #if (CHIPLET_AND_DIE == 0x1)
@@ -352,7 +350,7 @@ d1_dmac0: dma-controller-hsp@0x70430000 {
 			interrupt-parent = <&plic1>;
 			interrupts = <57>;
 			#dma-cells = <2>; // change dma-cells value <1> to <2>, to support peripheral selection dma-controller,See the parameter dmas for details;
-			clocks = <&d1_clock WIN2030_CLK_HSP_DMA0_CLK>;
+			clocks = <&d1_clock WIN2030_CLK_HSP_DMA0_CLK>,
 					 <&d1_clock WIN2030_CLK_HSP_DMA0_CLK_TEST>;
 			clock-names = "core-clk", "cfgr-clk";
 			resets = <&d1_reset HSPDMA_RST_CTRL SW_HSP_DMA0_RSTN>,
@@ -671,7 +669,7 @@ d1_ipc_scpu:ipc@1 {
 
 		d1_lpcpu:lpcpu@1 {
 			compatible = "eswin,win2030-lpcpu";
-			fw-region = <&lpcpu1_reserved>;
+			/* fw-region = <&lpcpu1_reserved>; */
 			clocks = <&d1_clock WIN2030_CLK_CLK_LPCPU_CORE>,
 					<&d1_clock WIN2030_CLK_CLK_LPCPU_BUS>;
 			clock-names = "core_clk", "bus_clk";
@@ -790,6 +788,7 @@ d1_i2c2: i2c@70970000 {
 				interrupts = <107>;
 				interrupt-parent = <&plic1>;
 				status = "disabled";
+				/*
 				d1_es8316: es8316@10 {
 					compatible = "everest,es8316";
 					reg = <0x10>;
@@ -803,6 +802,7 @@ d1_codec_endpoint: endpoint {
 						};
 					};
 				};
+				*/
 		};
 		d1_i2c3: i2c@70980000 {
 				compatible = "snps,designware-i2c";
@@ -968,7 +968,7 @@ d1_npu: eswin-npu@71c00000 {
 			dynamic-power-coefficient = <0>; /*TBD*/
 		};
 
-		dev_llc_d1: llc@71c00000 {
+		d1_llc_dev: llc@71c00000 {
 			compatible = "eswin,llc";
 			reg = <0x0 0x71c00000 0x0 0x400000>;
 			eswin,syscfg = <&d1_sys_con 0x324>;
@@ -1424,7 +1424,7 @@ dsp@0 {
 				};
 			};
 		};
-		die1_rtc: rtc@71818000 {
+		d1_rtc: rtc@71818000 {
 			compatible = "eswin,win2030-rtc";
 			reg = <0x0 0x71818000 0x0 0x400>;
 			eswin,syscfg = <&d1_sys_con 0x3c0>;
@@ -1438,7 +1438,7 @@ die1_rtc: rtc@71818000 {
 			status = "disabled";
 		};
 
-		pcie_die1: pcie@0x74000000 {
+		d0_pcie: pcie@0x74000000 {
 			compatible = "eswin,win2030-pcie";
 			clocks = <&d1_clock WIN2030_CLK_PCIET_ACLK>,
 				 <&d1_clock WIN2030_CLK_PCIET_CFG_CLK>,
diff --git a/arch/riscv/boot/dts/eswin/eswin-win2030-platform.dtsi b/arch/riscv/boot/dts/eswin/eswin-win2030-platform.dtsi
index 6d8817dfc2b6..0754fce5b9d7 100644
--- a/arch/riscv/boot/dts/eswin/eswin-win2030-platform.dtsi
+++ b/arch/riscv/boot/dts/eswin/eswin-win2030-platform.dtsi
@@ -31,7 +31,7 @@ BIT1  BIT0
 #define MEMMODE_FLAT
 #define AVAILABLE_DIE 0
 
-#define CHIPLET_AND_DIE		((CHIPLET_NUM << 1) | AVAILABLE_DIE)
+//#define CHIPLET_AND_DIE		((CHIPLET_NUM << 1) | AVAILABLE_DIE)
 
 /* VI related Macros */
 #define MIPI_CSI2_IMX290_ENBALE
diff --git a/arch/riscv/boot/dts/eswin/eswin-win2030.dts b/arch/riscv/boot/dts/eswin/eswin-win2030.dts
index 4f8c9f146ce3..8d9e0e76c350 100644
--- a/arch/riscv/boot/dts/eswin/eswin-win2030.dts
+++ b/arch/riscv/boot/dts/eswin/eswin-win2030.dts
@@ -1131,7 +1131,7 @@ &d1_dsp3 {
 	status = "disabled";
 };
 
-&pcie_die1 {
+&d0_pcie {
 	status = "disabled";
 };
 
@@ -1246,7 +1246,7 @@ &d1_timer3 {
 	status = "disabled";
 };
 
-&die1_rtc {
+&d1_rtc {
 	status = "disabled";
 };
 
diff --git a/arch/riscv/configs/eic7702_defconfig b/arch/riscv/configs/eic7702_defconfig
new file mode 100644
index 000000000000..36bcc734ee7e
--- /dev/null
+++ b/arch/riscv/configs/eic7702_defconfig
@@ -0,0 +1,834 @@
+CONFIG_LOCALVERSION="-eic7x"
+# CONFIG_LOCALVERSION_AUTO is not set
+CONFIG_SYSVIPC=y
+CONFIG_POSIX_MQUEUE=y
+CONFIG_WATCH_QUEUE=y
+CONFIG_NO_HZ_IDLE=y
+CONFIG_HIGH_RES_TIMERS=y
+CONFIG_BPF_SYSCALL=y
+CONFIG_PREEMPT_VOLUNTARY=y
+CONFIG_IKCONFIG=y
+CONFIG_IKCONFIG_PROC=y
+CONFIG_CGROUPS=y
+CONFIG_MEMCG=y
+CONFIG_BLK_CGROUP=y
+CONFIG_CGROUP_SCHED=y
+CONFIG_CFS_BANDWIDTH=y
+CONFIG_CGROUP_PIDS=y
+CONFIG_CGROUP_RDMA=y
+CONFIG_CGROUP_FREEZER=y
+CONFIG_CPUSETS=y
+CONFIG_CGROUP_DEVICE=y
+CONFIG_CGROUP_CPUACCT=y
+CONFIG_CGROUP_PERF=y
+CONFIG_CGROUP_BPF=y
+CONFIG_CGROUP_MISC=y
+CONFIG_NAMESPACES=y
+CONFIG_USER_NS=y
+CONFIG_CHECKPOINT_RESTORE=y
+CONFIG_EXPERT=y
+# CONFIG_SYSFS_SYSCALL is not set
+CONFIG_PERF_EVENTS=y
+CONFIG_SOC_SIFIVE=y
+CONFIG_SOC_VIRT=y
+CONFIG_SMP=y
+CONFIG_RISCV_SBI_V01=y
+# CONFIG_RISCV_BOOT_SPINWAIT is not set
+CONFIG_CMDLINE="earlycon=sbi console=tty1 console=ttyS0,115200n8 clk_ignore_unused cma_pernuma=0x2000000 disable_bypass=false no_console_suspend"
+CONFIG_CMDLINE_EXTEND=y
+CONFIG_PM_DEBUG=y
+CONFIG_PM_ADVANCED_DEBUG=y
+CONFIG_CPU_IDLE=y
+CONFIG_CPU_FREQ=y
+CONFIG_CPU_FREQ_STAT=y
+CONFIG_CPU_FREQ_DEFAULT_GOV_USERSPACE=y
+CONFIG_CPU_FREQ_GOV_PERFORMANCE=y
+CONFIG_CPU_FREQ_GOV_POWERSAVE=y
+CONFIG_CPU_FREQ_GOV_ONDEMAND=y
+CONFIG_CPU_FREQ_GOV_CONSERVATIVE=y
+CONFIG_CPU_FREQ_GOV_SCHEDUTIL=y
+CONFIG_CPUFREQ_DT=y
+CONFIG_VIRTUALIZATION=y
+CONFIG_KVM=m
+CONFIG_JUMP_LABEL=y
+# CONFIG_GCC_PLUGINS is not set
+CONFIG_MODULES=y
+CONFIG_MODULE_UNLOAD=y
+CONFIG_BLK_DEV_INTEGRITY=y
+CONFIG_BLK_DEV_THROTTLING=y
+CONFIG_BINFMT_MISC=m
+CONFIG_SPARSEMEM_MANUAL=y
+CONFIG_CMA=y
+CONFIG_NET=y
+CONFIG_PACKET=y
+CONFIG_PACKET_DIAG=m
+CONFIG_UNIX=y
+CONFIG_UNIX_DIAG=m
+CONFIG_TLS=m
+CONFIG_TLS_DEVICE=y
+CONFIG_TLS_TOE=y
+CONFIG_XDP_SOCKETS=y
+CONFIG_INET=y
+CONFIG_IP_MULTICAST=y
+CONFIG_IP_PNP=y
+CONFIG_IP_PNP_DHCP=y
+CONFIG_IP_PNP_BOOTP=y
+CONFIG_NET_IPIP=m
+CONFIG_NET_IPVTI=m
+CONFIG_TCP_CONG_ADVANCED=y
+CONFIG_TCP_CONG_HSTCP=m
+CONFIG_TCP_CONG_HYBLA=m
+CONFIG_TCP_CONG_NV=m
+CONFIG_TCP_CONG_SCALABLE=m
+CONFIG_TCP_CONG_LP=m
+CONFIG_TCP_CONG_VENO=m
+CONFIG_TCP_CONG_YEAH=m
+CONFIG_TCP_CONG_ILLINOIS=m
+CONFIG_TCP_CONG_DCTCP=m
+CONFIG_TCP_CONG_CDG=m
+CONFIG_TCP_CONG_BBR=y
+CONFIG_DEFAULT_BBR=y
+CONFIG_IPV6=m
+CONFIG_IPV6_ROUTER_PREF=y
+CONFIG_IPV6_OPTIMISTIC_DAD=y
+CONFIG_INET6_AH=m
+CONFIG_INET6_ESP=m
+CONFIG_INET6_IPCOMP=m
+CONFIG_IPV6_MIP6=m
+CONFIG_IPV6_ILA=m
+CONFIG_IPV6_VTI=m
+CONFIG_IPV6_MROUTE=y
+CONFIG_IPV6_SEG6_LWTUNNEL=y
+CONFIG_IPV6_SEG6_HMAC=y
+CONFIG_IPV6_RPL_LWTUNNEL=y
+CONFIG_IPV6_IOAM6_LWTUNNEL=y
+CONFIG_NETFILTER=y
+CONFIG_BRIDGE_NETFILTER=m
+CONFIG_NF_CONNTRACK=m
+CONFIG_NF_CONNTRACK_FTP=m
+CONFIG_NF_CONNTRACK_IRC=m
+CONFIG_NF_CONNTRACK_SNMP=m
+CONFIG_NF_CONNTRACK_PPTP=m
+CONFIG_NF_CONNTRACK_SIP=m
+CONFIG_NF_CONNTRACK_TFTP=m
+CONFIG_NF_CT_NETLINK=m
+CONFIG_NF_TABLES=m
+CONFIG_NF_TABLES_INET=y
+CONFIG_NF_TABLES_NETDEV=y
+CONFIG_NFT_NUMGEN=m
+CONFIG_NFT_CT=m
+CONFIG_NFT_CONNLIMIT=m
+CONFIG_NFT_LOG=m
+CONFIG_NFT_LIMIT=m
+CONFIG_NFT_MASQ=m
+CONFIG_NFT_REDIR=m
+CONFIG_NFT_NAT=m
+CONFIG_NFT_TUNNEL=m
+CONFIG_NFT_QUEUE=m
+CONFIG_NFT_QUOTA=m
+CONFIG_NFT_REJECT=m
+CONFIG_NFT_COMPAT=m
+CONFIG_NFT_HASH=m
+CONFIG_NFT_FIB_INET=m
+CONFIG_NFT_XFRM=m
+CONFIG_NFT_SOCKET=m
+CONFIG_NFT_OSF=m
+CONFIG_NFT_TPROXY=m
+CONFIG_NFT_SYNPROXY=m
+CONFIG_NFT_DUP_NETDEV=m
+CONFIG_NFT_FWD_NETDEV=m
+CONFIG_NFT_FIB_NETDEV=m
+CONFIG_NFT_REJECT_NETDEV=m
+CONFIG_NETFILTER_XT_TARGET_CHECKSUM=m
+CONFIG_NETFILTER_XT_TARGET_CLASSIFY=m
+CONFIG_NETFILTER_XT_TARGET_CONNMARK=m
+CONFIG_NETFILTER_XT_TARGET_CT=m
+CONFIG_NETFILTER_XT_TARGET_DSCP=m
+CONFIG_NETFILTER_XT_TARGET_HMARK=m
+CONFIG_NETFILTER_XT_TARGET_IDLETIMER=m
+CONFIG_NETFILTER_XT_TARGET_LED=m
+CONFIG_NETFILTER_XT_TARGET_LOG=m
+CONFIG_NETFILTER_XT_TARGET_MARK=m
+CONFIG_NETFILTER_XT_TARGET_NFLOG=m
+CONFIG_NETFILTER_XT_TARGET_NFQUEUE=m
+CONFIG_NETFILTER_XT_TARGET_TEE=m
+CONFIG_NETFILTER_XT_TARGET_TPROXY=m
+CONFIG_NETFILTER_XT_TARGET_TRACE=m
+CONFIG_NETFILTER_XT_TARGET_TCPMSS=m
+CONFIG_NETFILTER_XT_TARGET_TCPOPTSTRIP=m
+CONFIG_NETFILTER_XT_MATCH_ADDRTYPE=m
+CONFIG_NETFILTER_XT_MATCH_BPF=m
+CONFIG_NETFILTER_XT_MATCH_CGROUP=m
+CONFIG_NETFILTER_XT_MATCH_CLUSTER=m
+CONFIG_NETFILTER_XT_MATCH_COMMENT=m
+CONFIG_NETFILTER_XT_MATCH_CONNBYTES=m
+CONFIG_NETFILTER_XT_MATCH_CONNLABEL=m
+CONFIG_NETFILTER_XT_MATCH_CONNLIMIT=m
+CONFIG_NETFILTER_XT_MATCH_CONNMARK=m
+CONFIG_NETFILTER_XT_MATCH_CONNTRACK=m
+CONFIG_NETFILTER_XT_MATCH_CPU=m
+CONFIG_NETFILTER_XT_MATCH_DCCP=m
+CONFIG_NETFILTER_XT_MATCH_DEVGROUP=m
+CONFIG_NETFILTER_XT_MATCH_DSCP=m
+CONFIG_NETFILTER_XT_MATCH_ESP=m
+CONFIG_NETFILTER_XT_MATCH_HASHLIMIT=m
+CONFIG_NETFILTER_XT_MATCH_HELPER=m
+CONFIG_NETFILTER_XT_MATCH_IPCOMP=m
+CONFIG_NETFILTER_XT_MATCH_IPRANGE=m
+CONFIG_NETFILTER_XT_MATCH_IPVS=m
+CONFIG_NETFILTER_XT_MATCH_L2TP=m
+CONFIG_NETFILTER_XT_MATCH_LENGTH=m
+CONFIG_NETFILTER_XT_MATCH_LIMIT=m
+CONFIG_NETFILTER_XT_MATCH_MAC=m
+CONFIG_NETFILTER_XT_MATCH_MARK=m
+CONFIG_NETFILTER_XT_MATCH_MULTIPORT=m
+CONFIG_NETFILTER_XT_MATCH_NFACCT=m
+CONFIG_NETFILTER_XT_MATCH_OSF=m
+CONFIG_NETFILTER_XT_MATCH_OWNER=m
+CONFIG_NETFILTER_XT_MATCH_POLICY=m
+CONFIG_NETFILTER_XT_MATCH_PHYSDEV=m
+CONFIG_NETFILTER_XT_MATCH_PKTTYPE=m
+CONFIG_NETFILTER_XT_MATCH_QUOTA=m
+CONFIG_NETFILTER_XT_MATCH_RATEEST=m
+CONFIG_NETFILTER_XT_MATCH_REALM=m
+CONFIG_NETFILTER_XT_MATCH_RECENT=m
+CONFIG_NETFILTER_XT_MATCH_SCTP=m
+CONFIG_NETFILTER_XT_MATCH_SOCKET=m
+CONFIG_NETFILTER_XT_MATCH_STATE=m
+CONFIG_NETFILTER_XT_MATCH_STATISTIC=m
+CONFIG_NETFILTER_XT_MATCH_STRING=m
+CONFIG_NETFILTER_XT_MATCH_TCPMSS=m
+CONFIG_NETFILTER_XT_MATCH_TIME=m
+CONFIG_NETFILTER_XT_MATCH_U32=m
+CONFIG_IP_SET=m
+CONFIG_IP_VS=m
+CONFIG_IP_VS_IPV6=y
+CONFIG_IP_VS_PROTO_TCP=y
+CONFIG_IP_VS_PROTO_UDP=y
+CONFIG_IP_VS_PROTO_ESP=y
+CONFIG_IP_VS_RR=m
+CONFIG_IP_VS_NFCT=y
+CONFIG_NFT_DUP_IPV4=m
+CONFIG_NFT_FIB_IPV4=m
+CONFIG_NF_TABLES_ARP=y
+CONFIG_NF_LOG_ARP=m
+CONFIG_IP_NF_IPTABLES=m
+CONFIG_IP_NF_MATCH_AH=m
+CONFIG_IP_NF_MATCH_ECN=m
+CONFIG_IP_NF_MATCH_RPFILTER=m
+CONFIG_IP_NF_MATCH_TTL=m
+CONFIG_IP_NF_FILTER=m
+CONFIG_IP_NF_TARGET_REJECT=m
+CONFIG_IP_NF_TARGET_SYNPROXY=m
+CONFIG_IP_NF_NAT=m
+CONFIG_IP_NF_TARGET_MASQUERADE=m
+CONFIG_IP_NF_TARGET_NETMAP=m
+CONFIG_IP_NF_TARGET_REDIRECT=m
+CONFIG_IP_NF_MANGLE=m
+CONFIG_IP_NF_TARGET_ECN=m
+CONFIG_IP_NF_TARGET_TTL=m
+CONFIG_IP_NF_RAW=m
+CONFIG_IP_NF_ARPTABLES=m
+CONFIG_IP_NF_ARPFILTER=m
+CONFIG_IP_NF_ARP_MANGLE=m
+CONFIG_NFT_DUP_IPV6=m
+CONFIG_NFT_FIB_IPV6=m
+CONFIG_IP6_NF_IPTABLES=m
+CONFIG_IP6_NF_MATCH_AH=m
+CONFIG_IP6_NF_MATCH_EUI64=m
+CONFIG_IP6_NF_MATCH_FRAG=m
+CONFIG_IP6_NF_MATCH_OPTS=m
+CONFIG_IP6_NF_MATCH_HL=m
+CONFIG_IP6_NF_MATCH_IPV6HEADER=m
+CONFIG_IP6_NF_MATCH_MH=m
+CONFIG_IP6_NF_MATCH_RPFILTER=m
+CONFIG_IP6_NF_MATCH_RT=m
+CONFIG_IP6_NF_MATCH_SRH=m
+CONFIG_IP6_NF_TARGET_HL=m
+CONFIG_IP6_NF_FILTER=m
+CONFIG_IP6_NF_TARGET_REJECT=m
+CONFIG_IP6_NF_TARGET_SYNPROXY=m
+CONFIG_IP6_NF_MANGLE=m
+CONFIG_IP6_NF_RAW=m
+CONFIG_IP6_NF_NAT=m
+CONFIG_IP6_NF_TARGET_MASQUERADE=m
+CONFIG_IP6_NF_TARGET_NPT=m
+CONFIG_NF_TABLES_BRIDGE=m
+CONFIG_NF_CONNTRACK_BRIDGE=m
+CONFIG_BRIDGE_NF_EBTABLES=m
+CONFIG_BRIDGE_EBT_BROUTE=m
+CONFIG_BRIDGE_EBT_T_FILTER=m
+CONFIG_BRIDGE_EBT_T_NAT=m
+CONFIG_BRIDGE_EBT_802_3=m
+CONFIG_BRIDGE_EBT_AMONG=m
+CONFIG_BRIDGE_EBT_ARP=m
+CONFIG_BRIDGE_EBT_IP=m
+CONFIG_BRIDGE_EBT_IP6=m
+CONFIG_BRIDGE_EBT_LIMIT=m
+CONFIG_BRIDGE_EBT_MARK=m
+CONFIG_BRIDGE_EBT_PKTTYPE=m
+CONFIG_BRIDGE_EBT_STP=m
+CONFIG_BRIDGE_EBT_VLAN=m
+CONFIG_BRIDGE_EBT_ARPREPLY=m
+CONFIG_BRIDGE_EBT_DNAT=m
+CONFIG_BRIDGE_EBT_MARK_T=m
+CONFIG_BRIDGE_EBT_REDIRECT=m
+CONFIG_BRIDGE_EBT_SNAT=m
+CONFIG_BRIDGE_EBT_LOG=m
+CONFIG_BRIDGE_EBT_NFLOG=m
+CONFIG_BPFILTER=y
+CONFIG_BRIDGE=m
+CONFIG_BRIDGE_VLAN_FILTERING=y
+CONFIG_BRIDGE_MRP=y
+CONFIG_BRIDGE_CFM=y
+CONFIG_VLAN_8021Q=m
+CONFIG_VLAN_8021Q_GVRP=y
+CONFIG_LLC2=m
+CONFIG_NET_SCHED=y
+CONFIG_NET_CLS_CGROUP=m
+CONFIG_NET_CLS_ACT=y
+CONFIG_NETLINK_DIAG=m
+CONFIG_CGROUP_NET_PRIO=y
+CONFIG_BT=m
+CONFIG_BT_RFCOMM=m
+CONFIG_BT_BNEP=m
+CONFIG_BT_BNEP_MC_FILTER=y
+CONFIG_BT_BNEP_PROTO_FILTER=y
+CONFIG_BT_HIDP=m
+CONFIG_MCTP=y
+CONFIG_CFG80211=y
+CONFIG_CFG80211_WEXT=y
+CONFIG_MAC80211=y
+CONFIG_RFKILL=y
+CONFIG_RFKILL_INPUT=y
+CONFIG_RFKILL_GPIO=y
+CONFIG_PCI=y
+CONFIG_PCIEPORTBUS=y
+CONFIG_PCIEAER=y
+CONFIG_PCIEASPM_PERFORMANCE=y
+CONFIG_PCIE_PTM=y
+CONFIG_PCIE_ESWIN=y
+CONFIG_DEVTMPFS=y
+CONFIG_DEVTMPFS_MOUNT=y
+CONFIG_MTD=y
+CONFIG_MTD_SPI_NOR=y
+CONFIG_BLK_DEV_LOOP=y
+CONFIG_BLK_DEV_RAM=y
+CONFIG_BLK_DEV_RAM_COUNT=2
+CONFIG_BLK_DEV_RAM_SIZE=32768
+CONFIG_VIRTIO_BLK=y
+CONFIG_BLK_DEV_NVME=y
+CONFIG_NVME_MULTIPATH=y
+CONFIG_EEPROM_AT24=y
+# CONFIG_SCSI_PROC_FS is not set
+CONFIG_BLK_DEV_SD=y
+# CONFIG_BLK_DEV_BSG is not set
+# CONFIG_SCSI_LOWLEVEL is not set
+CONFIG_SCSI_DH=y
+CONFIG_SCSI_DH_RDAC=m
+CONFIG_SCSI_DH_EMC=m
+CONFIG_ATA=y
+CONFIG_SATA_AHCI=y
+CONFIG_AHCI_ESWIN=y
+CONFIG_MD=y
+CONFIG_BLK_DEV_DM=m
+CONFIG_DM_CRYPT=m
+CONFIG_DM_SNAPSHOT=m
+CONFIG_DM_THIN_PROVISIONING=m
+CONFIG_DM_WRITECACHE=m
+CONFIG_DM_MIRROR=m
+CONFIG_DM_RAID=m
+CONFIG_DM_MULTIPATH=m
+CONFIG_DM_MULTIPATH_QL=m
+CONFIG_DM_MULTIPATH_ST=m
+CONFIG_DM_MULTIPATH_HST=m
+CONFIG_DM_MULTIPATH_IOA=m
+CONFIG_DM_DELAY=m
+CONFIG_DM_VERITY=m
+CONFIG_NETDEVICES=y
+CONFIG_BONDING=m
+CONFIG_DUMMY=m
+CONFIG_WIREGUARD=m
+CONFIG_NET_FC=y
+CONFIG_NET_TEAM=m
+CONFIG_NET_TEAM_MODE_BROADCAST=m
+CONFIG_NET_TEAM_MODE_ROUNDROBIN=m
+CONFIG_NET_TEAM_MODE_RANDOM=m
+CONFIG_NET_TEAM_MODE_ACTIVEBACKUP=m
+CONFIG_NET_TEAM_MODE_LOADBALANCE=m
+CONFIG_MACVLAN=m
+CONFIG_MACVTAP=m
+CONFIG_IPVLAN=m
+CONFIG_IPVTAP=m
+CONFIG_VXLAN=m
+CONFIG_GENEVE=m
+CONFIG_VETH=m
+CONFIG_VIRTIO_NET=m
+# CONFIG_NET_VENDOR_ALACRITECH is not set
+# CONFIG_NET_VENDOR_AMAZON is not set
+# CONFIG_NET_VENDOR_AQUANTIA is not set
+# CONFIG_NET_VENDOR_ARC is not set
+# CONFIG_NET_VENDOR_ASIX is not set
+# CONFIG_NET_VENDOR_BROADCOM is not set
+# CONFIG_NET_VENDOR_CADENCE is not set
+# CONFIG_NET_VENDOR_CAVIUM is not set
+# CONFIG_NET_VENDOR_CORTINA is not set
+# CONFIG_NET_VENDOR_ENGLEDER is not set
+# CONFIG_NET_VENDOR_EZCHIP is not set
+# CONFIG_NET_VENDOR_GOOGLE is not set
+# CONFIG_NET_VENDOR_HUAWEI is not set
+# CONFIG_NET_VENDOR_INTEL is not set
+# CONFIG_NET_VENDOR_LITEX is not set
+# CONFIG_NET_VENDOR_MARVELL is not set
+# CONFIG_NET_VENDOR_MELLANOX is not set
+# CONFIG_NET_VENDOR_MICREL is not set
+# CONFIG_NET_VENDOR_MICROCHIP is not set
+# CONFIG_NET_VENDOR_MICROSEMI is not set
+# CONFIG_NET_VENDOR_MICROSOFT is not set
+# CONFIG_NET_VENDOR_NI is not set
+# CONFIG_NET_VENDOR_NATSEMI is not set
+# CONFIG_NET_VENDOR_NETRONOME is not set
+# CONFIG_NET_VENDOR_PENSANDO is not set
+# CONFIG_NET_VENDOR_QUALCOMM is not set
+# CONFIG_NET_VENDOR_RENESAS is not set
+# CONFIG_NET_VENDOR_ROCKER is not set
+# CONFIG_NET_VENDOR_SAMSUNG is not set
+# CONFIG_NET_VENDOR_SEEQ is not set
+# CONFIG_NET_VENDOR_SOLARFLARE is not set
+# CONFIG_NET_VENDOR_SOCIONEXT is not set
+CONFIG_STMMAC_ETH=y
+CONFIG_STMMAC_SELFTESTS=y
+CONFIG_DWMAC_WIN2030=y
+# CONFIG_DWMAC_GENERIC is not set
+# CONFIG_NET_VENDOR_SYNOPSYS is not set
+# CONFIG_NET_VENDOR_VERTEXCOM is not set
+# CONFIG_NET_VENDOR_VIA is not set
+# CONFIG_NET_VENDOR_WIZNET is not set
+# CONFIG_NET_VENDOR_XILINX is not set
+# CONFIG_WLAN_VENDOR_ADMTEK is not set
+# CONFIG_WLAN_VENDOR_ATH is not set
+# CONFIG_WLAN_VENDOR_ATMEL is not set
+# CONFIG_WLAN_VENDOR_BROADCOM is not set
+# CONFIG_WLAN_VENDOR_CISCO is not set
+# CONFIG_WLAN_VENDOR_INTEL is not set
+# CONFIG_WLAN_VENDOR_INTERSIL is not set
+# CONFIG_WLAN_VENDOR_MARVELL is not set
+# CONFIG_WLAN_VENDOR_MEDIATEK is not set
+# CONFIG_WLAN_VENDOR_MICROCHIP is not set
+# CONFIG_WLAN_VENDOR_PURELIFI is not set
+# CONFIG_WLAN_VENDOR_RALINK is not set
+CONFIG_RTL8192SE=m
+# CONFIG_WLAN_VENDOR_RSI is not set
+# CONFIG_WLAN_VENDOR_SILABS is not set
+# CONFIG_WLAN_VENDOR_ST is not set
+# CONFIG_WLAN_VENDOR_TI is not set
+# CONFIG_WLAN_VENDOR_ZYDAS is not set
+# CONFIG_WLAN_VENDOR_QUANTENNA is not set
+CONFIG_INPUT_EVDEV=y
+# CONFIG_INPUT_KEYBOARD is not set
+# CONFIG_INPUT_MOUSE is not set
+CONFIG_SERIO_LIBPS2=y
+# CONFIG_LEGACY_PTYS is not set
+CONFIG_SERIAL_8250=y
+CONFIG_SERIAL_8250_CONSOLE=y
+CONFIG_SERIAL_8250_DW=y
+CONFIG_SERIAL_OF_PLATFORM=y
+CONFIG_SERIAL_EARLYCON_RISCV_SBI=y
+CONFIG_HVC_RISCV_SBI=y
+CONFIG_VIRTIO_CONSOLE=y
+CONFIG_HW_RANDOM=y
+CONFIG_HW_RANDOM_VIRTIO=y
+# CONFIG_I2C_COMPAT is not set
+CONFIG_I2C_CHARDEV=y
+# CONFIG_I2C_HELPER_AUTO is not set
+CONFIG_I2C_DESIGNWARE_SLAVE=y
+CONFIG_I2C_DESIGNWARE_PLATFORM=y
+CONFIG_I2C_SLAVE_EEPROM=y
+CONFIG_SPI=y
+CONFIG_SPI_DESIGNWARE=y
+CONFIG_SPI_DW_DMA=y
+CONFIG_SPI_DW_MMIO=y
+CONFIG_SPI_ESWIN_BOOTSPI=y
+# CONFIG_PTP_1588_CLOCK is not set
+CONFIG_PINCTRL=y
+CONFIG_PINCTRL_EIC7700=y
+CONFIG_GPIOLIB=y
+CONFIG_GPIO_SYSFS=y
+CONFIG_GPIO_DWAPB=y
+CONFIG_SENSORS_ESWIN_FAN_CONTROL=y
+CONFIG_SENSORS_ESWIN_PVT=y
+CONFIG_SENSORS_INA2XX=y
+CONFIG_SENSORS_PAC1934=y
+CONFIG_WATCHDOG=y
+CONFIG_DW_WATCHDOG=y
+CONFIG_REGULATOR=y
+CONFIG_REGULATOR_MPQ8785=y
+CONFIG_REGULATOR_PCA9450=y
+CONFIG_MEDIA_SUPPORT=y
+CONFIG_MEDIA_SUPPORT_FILTER=y
+CONFIG_MEDIA_CAMERA_SUPPORT=y
+CONFIG_MEDIA_ANALOG_TV_SUPPORT=y
+CONFIG_MEDIA_DIGITAL_TV_SUPPORT=y
+CONFIG_MEDIA_RADIO_SUPPORT=y
+CONFIG_MEDIA_SDR_SUPPORT=y
+CONFIG_MEDIA_PLATFORM_SUPPORT=y
+CONFIG_MEDIA_USB_SUPPORT=y
+CONFIG_USB_PWC=m
+CONFIG_USB_VIDEO_CLASS=m
+CONFIG_MEDIA_PCI_SUPPORT=y
+CONFIG_V4L_PLATFORM_DRIVERS=y
+CONFIG_V4L_MEM2MEM_DRIVERS=y
+CONFIG_VIDEO_XILINX=y
+CONFIG_CXD2880_SPI_DRV=m
+CONFIG_MEDIA_TUNER_E4000=m
+CONFIG_MEDIA_TUNER_FC0011=m
+CONFIG_MEDIA_TUNER_FC0012=m
+CONFIG_MEDIA_TUNER_FC0013=m
+CONFIG_MEDIA_TUNER_FC2580=m
+CONFIG_MEDIA_TUNER_IT913X=m
+CONFIG_MEDIA_TUNER_M88RS6000T=m
+CONFIG_MEDIA_TUNER_MAX2165=m
+CONFIG_MEDIA_TUNER_MSI001=m
+CONFIG_MEDIA_TUNER_MT2060=m
+CONFIG_MEDIA_TUNER_MT2063=m
+CONFIG_MEDIA_TUNER_MT2131=m
+CONFIG_MEDIA_TUNER_MT2266=m
+CONFIG_MEDIA_TUNER_MXL301RF=m
+CONFIG_MEDIA_TUNER_MXL5005S=m
+CONFIG_MEDIA_TUNER_MXL5007T=m
+CONFIG_MEDIA_TUNER_QM1D1B0004=m
+CONFIG_MEDIA_TUNER_QM1D1C0042=m
+CONFIG_MEDIA_TUNER_QT1010=m
+CONFIG_MEDIA_TUNER_R820T=m
+CONFIG_MEDIA_TUNER_SI2157=m
+CONFIG_MEDIA_TUNER_TDA18212=m
+CONFIG_MEDIA_TUNER_TDA18218=m
+CONFIG_MEDIA_TUNER_TDA18250=m
+CONFIG_MEDIA_TUNER_TUA9001=m
+CONFIG_DVB_MXL5XX=m
+CONFIG_DVB_STB0899=m
+CONFIG_DVB_STB6100=m
+CONFIG_DVB_STV090x=m
+CONFIG_DVB_STV0910=m
+CONFIG_DVB_STV6110x=m
+CONFIG_DVB_STV6111=m
+CONFIG_DVB_DRXK=m
+CONFIG_DVB_MN88472=m
+CONFIG_DVB_MN88473=m
+CONFIG_DVB_SI2165=m
+CONFIG_DVB_TDA18271C2DD=m
+CONFIG_DVB_CX24110=m
+CONFIG_DVB_CX24116=m
+CONFIG_DVB_CX24117=m
+CONFIG_DVB_CX24120=m
+CONFIG_DVB_CX24123=m
+CONFIG_DVB_DS3000=m
+CONFIG_DVB_MB86A16=m
+CONFIG_DVB_MT312=m
+CONFIG_DVB_S5H1420=m
+CONFIG_DVB_SI21XX=m
+CONFIG_DVB_STB6000=m
+CONFIG_DVB_STV0288=m
+CONFIG_DVB_STV0299=m
+CONFIG_DVB_STV0900=m
+CONFIG_DVB_STV6110=m
+CONFIG_DVB_TDA10071=m
+CONFIG_DVB_TDA10086=m
+CONFIG_DVB_TDA8083=m
+CONFIG_DVB_TDA8261=m
+CONFIG_DVB_TDA826X=m
+CONFIG_DVB_TS2020=m
+CONFIG_DVB_TUA6100=m
+CONFIG_DVB_TUNER_CX24113=m
+CONFIG_DVB_TUNER_ITD1000=m
+CONFIG_DVB_VES1X93=m
+CONFIG_DVB_ZL10036=m
+CONFIG_DVB_ZL10039=m
+CONFIG_DVB_CX22700=m
+CONFIG_DVB_CX22702=m
+CONFIG_DVB_CXD2820R=m
+CONFIG_DVB_CXD2841ER=m
+CONFIG_DVB_DIB3000MB=m
+CONFIG_DVB_DIB3000MC=m
+CONFIG_DVB_DIB7000M=m
+CONFIG_DVB_DIB7000P=m
+CONFIG_DVB_DIB9000=m
+CONFIG_DVB_DRXD=m
+CONFIG_DVB_EC100=m
+CONFIG_DVB_L64781=m
+CONFIG_DVB_MT352=m
+CONFIG_DVB_NXT6000=m
+CONFIG_DVB_S5H1432=m
+CONFIG_DVB_SP887X=m
+CONFIG_DVB_STV0367=m
+CONFIG_DVB_TDA10048=m
+CONFIG_DVB_TDA1004X=m
+CONFIG_DVB_ZD1301_DEMOD=m
+CONFIG_DVB_ZL10353=m
+CONFIG_DVB_CXD2880=m
+CONFIG_DVB_STV0297=m
+CONFIG_DVB_TDA10021=m
+CONFIG_DVB_TDA10023=m
+CONFIG_DVB_VES1820=m
+CONFIG_DVB_AU8522_DTV=m
+CONFIG_DVB_AU8522_V4L=m
+CONFIG_DVB_BCM3510=m
+CONFIG_DVB_LG2160=m
+CONFIG_DVB_LGDT3305=m
+CONFIG_DVB_LGDT330X=m
+CONFIG_DVB_MXL692=m
+CONFIG_DVB_NXT200X=m
+CONFIG_DVB_OR51132=m
+CONFIG_DVB_OR51211=m
+CONFIG_DVB_S5H1409=m
+CONFIG_DVB_S5H1411=m
+CONFIG_DVB_DIB8000=m
+CONFIG_DVB_MB86A20S=m
+CONFIG_DVB_S921=m
+CONFIG_DVB_MN88443X=m
+CONFIG_DVB_TC90522=m
+CONFIG_DVB_PLL=m
+CONFIG_DVB_TUNER_DIB0070=m
+CONFIG_DVB_TUNER_DIB0090=m
+CONFIG_DVB_A8293=m
+CONFIG_DVB_AF9033=m
+CONFIG_DVB_ASCOT2E=m
+CONFIG_DVB_ATBM8830=m
+CONFIG_DVB_HELENE=m
+CONFIG_DVB_HORUS3A=m
+CONFIG_DVB_ISL6405=m
+CONFIG_DVB_ISL6421=m
+CONFIG_DVB_ISL6423=m
+CONFIG_DVB_IX2505V=m
+CONFIG_DVB_LGS8GL5=m
+CONFIG_DVB_LGS8GXX=m
+CONFIG_DVB_LNBH25=m
+CONFIG_DVB_LNBH29=m
+CONFIG_DVB_LNBP21=m
+CONFIG_DVB_LNBP22=m
+CONFIG_DVB_M88RS2000=m
+CONFIG_DVB_TDA665x=m
+CONFIG_DVB_DRX39XYJ=m
+CONFIG_DVB_CXD2099=m
+CONFIG_DVB_SP2=m
+# CONFIG_DRM_I2C_CH7006 is not set
+# CONFIG_DRM_I2C_SIL164 is not set
+CONFIG_DRM_I2C_NXP_TDA9950=y
+CONFIG_DRM_RADEON=m
+CONFIG_DRM_AMDGPU=m
+CONFIG_DRM_AMDGPU_SI=y
+CONFIG_DRM_AMDGPU_CIK=y
+CONFIG_DRM_AMDGPU_USERPTR=y
+CONFIG_DRM_AMD_ACP=y
+CONFIG_DRM_AMD_DC_SI=y
+CONFIG_DRM_NOUVEAU=y
+CONFIG_DRM_DISPLAY_CONNECTOR=y
+CONFIG_DRM_SIMPLE_BRIDGE=y
+CONFIG_DRM_TOSHIBA_TC358768=m
+CONFIG_DRM_SIMPLEDRM=m
+CONFIG_DRM_ESWIN=y
+CONFIG_ESWIN_MMU=y
+CONFIG_ESWIN_DW_HDMI=y
+CONFIG_DW_HDMI_I2S_AUDIO=y
+CONFIG_DW_HDMI_CEC=y
+CONFIG_DRM_IMG_VOLCANIC=m
+CONFIG_DRM_LEGACY=y
+CONFIG_FB=y
+CONFIG_FB_RADEON=m
+CONFIG_FB_SIMPLE=m
+CONFIG_SOUND=y
+CONFIG_SND=y
+CONFIG_SND_HDA_INTEL=m
+CONFIG_SND_HDA_PATCH_LOADER=y
+CONFIG_SND_HDA_CODEC_REALTEK=m
+CONFIG_SND_HDA_CODEC_HDMI=m
+CONFIG_SND_HDA_PREALLOC_SIZE=2048
+CONFIG_SND_SOC=y
+CONFIG_SND_SOC_SOF_TOPLEVEL=y
+CONFIG_SND_SOC_SOF_OF=y
+CONFIG_SND_ESWIN_DW_I2S=y
+CONFIG_ESWIN_SND_SOC_CODECS=y
+CONFIG_ESWIN_SND_ES8388_CODEC=y
+CONFIG_SND_SIMPLE_CARD=y
+CONFIG_SND_AUDIO_GRAPH_CARD=y
+CONFIG_USB_ULPI_BUS=y
+CONFIG_USB_CONN_GPIO=y
+CONFIG_USB=y
+CONFIG_USB_ANNOUNCE_NEW_DEVICES=y
+# CONFIG_USB_DEFAULT_PERSIST is not set
+CONFIG_USB_MON=y
+CONFIG_USB_XHCI_HCD=y
+CONFIG_USB_XHCI_DBGCAP=y
+CONFIG_USB_EHCI_HCD=y
+CONFIG_USB_EHCI_ROOT_HUB_TT=y
+CONFIG_USB_EHCI_HCD_PLATFORM=y
+CONFIG_USB_OHCI_HCD=y
+CONFIG_USB_OHCI_HCD_PLATFORM=y
+CONFIG_USB_ACM=y
+CONFIG_USB_WDM=y
+CONFIG_USB_STORAGE=y
+CONFIG_USB_UAS=y
+CONFIG_USB_DWC3=y
+CONFIG_USB_DWC3_ULPI=y
+CONFIG_USB_DWC2=y
+CONFIG_USB_SERIAL=y
+CONFIG_USB_SERIAL_GENERIC=y
+CONFIG_USB_SERIAL_CP210X=y
+CONFIG_USB_SERIAL_FTDI_SIO=y
+CONFIG_USB_SERIAL_KEYSPAN=y
+CONFIG_USB_SERIAL_PL2303=y
+CONFIG_USB_SERIAL_OTI6858=y
+CONFIG_USB_SERIAL_QUALCOMM=y
+CONFIG_USB_SERIAL_OPTION=y
+CONFIG_USB_TEST=y
+CONFIG_USB_GADGET=y
+CONFIG_USB_GADGET_DEBUG_FILES=y
+CONFIG_USB_GADGET_VBUS_DRAW=500
+CONFIG_USB_CONFIGFS=y
+CONFIG_USB_CONFIGFS_MASS_STORAGE=y
+CONFIG_USB_CONFIGFS_F_LB_SS=y
+CONFIG_USB_CONFIGFS_F_FS=y
+CONFIG_USB_ZERO=m
+CONFIG_USB_MASS_STORAGE=m
+CONFIG_TYPEC=y
+CONFIG_TYPEC_FUSB303B=y
+CONFIG_MMC=y
+CONFIG_MMC_TEST=y
+CONFIG_MMC_DEBUG=y
+CONFIG_MMC_SDHCI=y
+CONFIG_MMC_SDHCI_PLTFM=y
+CONFIG_MMC_SDHCI_OF_ESWIN=y
+CONFIG_MMC_SDHCI_OF_SDIO_ESWIN=y
+CONFIG_LEDS_CLASS=y
+CONFIG_LEDS_GPIO=y
+CONFIG_LEDS_TRIGGER_TIMER=y
+CONFIG_LEDS_TRIGGER_ONESHOT=y
+CONFIG_LEDS_TRIGGER_HEARTBEAT=y
+CONFIG_LEDS_TRIGGER_BACKLIGHT=y
+CONFIG_LEDS_TRIGGER_DEFAULT_ON=y
+CONFIG_EDAC=y
+CONFIG_EDAC_ESWIN=y
+CONFIG_RTC_CLASS=y
+CONFIG_RTC_DRV_PCF8563=y
+CONFIG_RTC_DRV_ESWIN=y
+CONFIG_DMADEVICES=y
+CONFIG_DW_AXI_DMAC=y
+CONFIG_DMATEST=m
+CONFIG_DMABUF_HEAPS=y
+CONFIG_DMABUF_HEAPS_SYSTEM=y
+CONFIG_DMABUF_HEAPS_CMA=y
+CONFIG_VIRTIO_BALLOON=y
+CONFIG_VIRTIO_INPUT=y
+CONFIG_VIRTIO_MMIO=y
+CONFIG_STAGING=y
+CONFIG_STAGING_MEDIA=y
+CONFIG_VIDEO_ESWIN=y
+CONFIG_VIDEO_ESWIN_VDEC=y
+CONFIG_VIDEO_ESWIN_VENC=y
+CONFIG_VIDEO_ESWIN_HAE=y
+CONFIG_VIDEO_ESWIN_DEWARP=y
+CONFIG_VIDEO_ESWIN_MEDIA_EXT_DRIVER=y
+CONFIG_COMMON_CLK_WIN2030=y
+CONFIG_TIMER_ESWIN=y
+CONFIG_MAILBOX=y
+CONFIG_ESWIN_MBOX=y
+CONFIG_ESWIN_LPCPU=m
+CONFIG_ARM_SMMU_V3=y
+CONFIG_RPMSG_VIRTIO=y
+CONFIG_ARCH_ESWIN_EIC770X_SOC_FAMILY=y
+CONFIG_ARCH_ESWIN_EIC7702_SOC=y
+CONFIG_EXTCON=y
+CONFIG_MEMORY=y
+CONFIG_ESWIN_BUDDY=y
+CONFIG_ESWIN_PROC=y
+CONFIG_ESWIN_RSVMEM_HEAP=y
+CONFIG_ESWIN_MMZ_VB=y
+CONFIG_ESWIN_DEV_DMA_BUF=y
+CONFIG_ESWIN_IOMMU_RSV=y
+CONFIG_PWM=y
+CONFIG_PWM_ESWIN=y
+CONFIG_RESET_ESWIN_WIN2030=y
+CONFIG_INTERCONNECT=y
+CONFIG_EXT4_FS=y
+CONFIG_EXT4_FS_POSIX_ACL=y
+CONFIG_EXT4_FS_SECURITY=y
+CONFIG_BTRFS_FS=m
+CONFIG_BTRFS_FS_POSIX_ACL=y
+CONFIG_FS_ENCRYPTION=y
+CONFIG_FANOTIFY=y
+CONFIG_OVERLAY_FS=m
+CONFIG_OVERLAY_FS_REDIRECT_DIR=y
+CONFIG_OVERLAY_FS_INDEX=y
+CONFIG_OVERLAY_FS_XINO_AUTO=y
+CONFIG_ISO9660_FS=m
+CONFIG_JOLIET=y
+CONFIG_ZISOFS=y
+CONFIG_MSDOS_FS=y
+CONFIG_VFAT_FS=y
+CONFIG_EXFAT_FS=m
+CONFIG_NTFS_FS=m
+CONFIG_NTFS_RW=y
+CONFIG_TMPFS=y
+CONFIG_TMPFS_POSIX_ACL=y
+CONFIG_SQUASHFS=m
+CONFIG_SQUASHFS_FILE_DIRECT=y
+CONFIG_SQUASHFS_COMPILE_DECOMP_MULTI_PERCPU=y
+CONFIG_SQUASHFS_XATTR=y
+CONFIG_SQUASHFS_LZ4=y
+CONFIG_SQUASHFS_LZO=y
+CONFIG_SQUASHFS_XZ=y
+CONFIG_SQUASHFS_ZSTD=y
+CONFIG_NFS_FS=y
+CONFIG_NFS_V3_ACL=y
+CONFIG_NFS_V4=y
+CONFIG_NFS_V4_1=y
+CONFIG_NFS_V4_2=y
+CONFIG_ROOT_NFS=y
+CONFIG_NLS_CODEPAGE_437=y
+CONFIG_NLS_ISO8859_1=m
+CONFIG_SECURITY=y
+CONFIG_SECURITY_SELINUX=y
+CONFIG_SECURITY_SELINUX_BOOTPARAM=y
+CONFIG_SECURITY_APPARMOR=y
+CONFIG_DEFAULT_SECURITY_APPARMOR=y
+CONFIG_LSM="landlock,lockdown,yama,integrity,apparmor"
+CONFIG_CRYPTO_CBC=m
+# CONFIG_CRYPTO_CTS is not set
+CONFIG_CRYPTO_ECB=m
+# CONFIG_CRYPTO_XTS is not set
+CONFIG_CRYPTO_HMAC=m
+CONFIG_CRYPTO_MD5=y
+CONFIG_CRYPTO_SHA512=m
+CONFIG_CRYPTO_DEV_VIRTIO=y
+CONFIG_CRC_ITU_T=y
+CONFIG_CRC7=y
+CONFIG_XZ_DEC=y
+CONFIG_DMA_CMA=y
+CONFIG_PRINTK_TIME=y
+CONFIG_CONSOLE_LOGLEVEL_DEFAULT=15
+CONFIG_CONSOLE_LOGLEVEL_QUIET=15
+CONFIG_MESSAGE_LOGLEVEL_DEFAULT=7
+CONFIG_DYNAMIC_DEBUG=y
+CONFIG_DEBUG_FS=y
+CONFIG_DEBUG_PAGEALLOC=y
+CONFIG_SCHED_STACK_END_CHECK=y
+CONFIG_DEBUG_VM=y
+CONFIG_DEBUG_VM_PGFLAGS=y
+CONFIG_DEBUG_MEMORY_INIT=y
+CONFIG_DEBUG_PER_CPU_MAPS=y
+CONFIG_SOFTLOCKUP_DETECTOR=y
+CONFIG_WQ_WATCHDOG=y
+CONFIG_DEBUG_TIMEKEEPING=y
+CONFIG_DEBUG_RT_MUTEXES=y
+CONFIG_DEBUG_SPINLOCK=y
+CONFIG_DEBUG_MUTEXES=y
+CONFIG_DEBUG_RWSEMS=y
+CONFIG_DEBUG_ATOMIC_SLEEP=y
+CONFIG_DEBUG_LIST=y
+CONFIG_DEBUG_PLIST=y
+CONFIG_DEBUG_SG=y
+# CONFIG_RCU_TRACE is not set
+CONFIG_RCU_EQS_DEBUG=y
+# CONFIG_FTRACE is not set
+# CONFIG_RUNTIME_TESTING_MENU is not set
+CONFIG_MEMTEST=y
-- 
2.49.0

